# Caderno de Pesquisa 

### Canais de Youtube da família Bolsonaro

- Carlos
- Eduardo
- Jair
- Flavio

### Dias

- Carlos (22/07/20)
- Eduardo (23/07/20)
- Jair (23/07/20)
- Flávio (24/07/20)

### Percurso

Nesse primeiro momento busca-se leantar os dados abaixo. 
Tenho como ponto principal observar vídeos produzidos/publicados para os canais de cunho humorístico.
Nesse momento que observar se há vídeos assim e como são. 

1. 50 VIDS
2. Todos VIDS
3. Stats do canal 
4. Comentários de alguns vídeos

**Referência** *a serem consideradas: fatos noticiados sobre a família e o YouTube!*

https://jc.ne10.uol.com.br/politica/2020/05/5610657-saiba-quem-sao-os-bolsonaristas-alvos-da-policia-federal-na-investigacao-das-fake-news.html

https://theintercept.com/2018/11/17/youtubers-bolsonaro-nando-moura-diego-rox-bernardo-kuster-fake-news/


## Bibliotecas

In [1]:
import pandas as pd
from matplotlib import pyplot as plt
from datetime import datetime
from googleapiclient.discovery import build

## Configurar - api YouTube V3##

**api_key** = " [inserir o seu api_key] "

In [ ]:
youtube = build("youtube", "v3", developerKey=api_key)

<h1>Carlos</h1>

In [8]:
req_carlosb = youtube.channels().list(id = "UCBrntWupyTRnnsRy3dA4VdA'", part = "contentDetails")
canalcarlosb = req_carlosb.execute()

In [10]:
canalcarlosb

{'kind': 'youtube#channelListResponse',
 'etag': 'HFuuexSA9gsOV5Sf-9KlMwRoT50',
 'pageInfo': {'totalResults': 1, 'resultsPerPage': 1},
 'items': [{'kind': 'youtube#channel',
   'etag': '76HWfqm-NisrPPUqmK02QKtMDU0',
   'id': 'UCBrntWupyTRnnsRy3dA4VdA',
   'contentDetails': {'relatedPlaylists': {'likes': '',
     'favorites': '',
     'uploads': 'UUBrntWupyTRnnsRy3dA4VdA',
     'watchHistory': 'HL',
     'watchLater': 'WL'}}}]}

In [12]:
#'uploads': 'UUBrntWupyTRnnsRy3dA4VdA'

req_playlist = youtube.playlistItems().list(playlistId = "UUBrntWupyTRnnsRy3dA4VdA", 
                                            part = "snippet", 
                                            maxResults = 50)

carlos_playlist = req_playlist.execute()
                                        

In [17]:
carlos_playlist['items'][0]

{'kind': 'youtube#playlistItem',
 'etag': 'dsPaQ3hw-IljxqVpFJHAuOSMsfY',
 'id': 'VVVCcm50V3VweVRSbm5zUnkzZEE0VmRBLm9WeEZGcFpqaWdn',
 'snippet': {'publishedAt': '2020-07-21T10:14:05Z',
  'channelId': 'UCBrntWupyTRnnsRy3dA4VdA',
  'title': 'Aeroportos de Congonhas e Navegantes passam por reformas para o bem estar do usuário.',
  'description': '',
  'thumbnails': {'default': {'url': 'https://i.ytimg.com/vi/oVxFFpZjigg/default.jpg',
    'width': 120,
    'height': 90},
   'medium': {'url': 'https://i.ytimg.com/vi/oVxFFpZjigg/mqdefault.jpg',
    'width': 320,
    'height': 180},
   'high': {'url': 'https://i.ytimg.com/vi/oVxFFpZjigg/hqdefault.jpg',
    'width': 480,
    'height': 360},
   'standard': {'url': 'https://i.ytimg.com/vi/oVxFFpZjigg/sddefault.jpg',
    'width': 640,
    'height': 480}},
  'channelTitle': 'Carlos Bolsonaro',
  'playlistId': 'UUBrntWupyTRnnsRy3dA4VdA',
  'position': 0,
  'resourceId': {'kind': 'youtube#video', 'videoId': 'oVxFFpZjigg'}}}

**Os 50 vídeos de destaque do canal**

In [18]:
# pegar 50 vídeo = ['title'][link]    
for r in carlos_playlist['items']:
    print(r['snippet']['title'], "https://www.youtube.com/watch?v="+r['snippet']['resourceId']['videoId'])

Aeroportos de Congonhas e Navegantes passam por reformas para o bem estar do usuário. https://www.youtube.com/watch?v=oVxFFpZjigg
Ferrovia Oeste-Leste será entregue em 2022 com mais 485,4km de extensão. Detalhes na descrição: https://www.youtube.com/watch?v=W12HyI8hrvk
Duplicação da BR-101/NE está sendo feita em pavimento rígido (22 cm + 10 cm de concreto compactado) https://www.youtube.com/watch?v=EhBiTP7QDBo
De um Pernambucano ao Presidente Bolsonaro. Por isso querem tanto calar a internet, a voz do povo! https://www.youtube.com/watch?v=PDhyGCql-_0
Plano emergencial de enfrentamento à Covid19 / Operação Acolhida em Boa Vista (RR). https://www.youtube.com/watch?v=yx8N8OCDyvc
Quando esquecem de combinar com a entrevistada. https://www.youtube.com/watch?v=jVbrSEP7NKI
Fase de conclusão do Complexo Viário em Blumenau e mais BRs. Detalhes de mais obras na descrição: https://www.youtube.com/watch?v=BFx1Saet9dA
Governo do Brasil retorna dragagem do Rio Madeira. Mais detalhes na descrição: ht

In [20]:
for r in carlos_playlist['items']:
    print(r['snippet']['title'], r['snippet']['description'], "https://www.youtube.com/watch?v="+r['snippet']['resourceId']['videoId'])

Aeroportos de Congonhas e Navegantes passam por reformas para o bem estar do usuário.  https://www.youtube.com/watch?v=oVxFFpZjigg
Ferrovia Oeste-Leste será entregue em 2022 com mais 485,4km de extensão. Detalhes na descrição: O Governo do Brasil  e o Ministério da Infraestrutura priorizam a construção da Ferrovia de Integração Oeste-Leste/Fiol! São mais de mil colaboradores para entregar o 2º trecho da  ferrovia em 2022. Com 485,4 km de extensão, o trecho 2, entre Caetité/BA e Barreiras/BA conectará a região produtora de grãos do oeste da Bahia ao porto de Ilhéus. https://www.youtube.com/watch?v=W12HyI8hrvk
Duplicação da BR-101/NE está sendo feita em pavimento rígido (22 cm + 10 cm de concreto compactado) ‪- Duplicação da BR-101/NE está sendo feita em pavimento rígido (22 cm + 10 cm de concreto compactado).‬

‪- Como determinado pelo Capitão Tarcísio  , Ministro da Infraestrutura , o mesmo serviço está sendo feito na BR-285/SC.‬ https://www.youtube.com/watch?v=EhBiTP7QDBo
De um Pernam

**Salvar**

In [31]:

Titulos_CB = []

for r in carlos_playlist['items']:
    Title = r['snippet']['title']
    Coment = r['snippet']['description']
    Link = ("https://www.youtube.com/watch?v="+r['snippet']['resourceId']['videoId'])
    Titulos_CB.append((Title,Coment, Link))

Titulos_CB = pd.DataFrame(Titulos_CB, columns=['Titulo', 'Comentário', 'Link'])

Titulos_CB.head()

Titulo  \
0  Aeroportos de Congonhas e Navegantes passam po...   
1  Ferrovia Oeste-Leste será entregue em 2022 com...   
2  Duplicação da BR-101/NE está sendo feita em pa...   
3  De um Pernambucano ao Presidente Bolsonaro. Po...   
4  Plano emergencial de enfrentamento à Covid19 /...   

                                          Comentário  \
0                                                      
1  O Governo do Brasil  e o Ministério da Infraes...   
2  ‪- Duplicação da BR-101/NE está sendo feita em...   
3                                                      
4                                                      

                                          Link  
0  https://www.youtube.com/watch?v=oVxFFpZjigg  
1  https://www.youtube.com/watch?v=W12HyI8hrvk  
2  https://www.youtube.com/watch?v=EhBiTP7QDBo  
3  https://www.youtube.com/watch?v=PDhyGCql-_0  
4  https://www.youtube.com/watch?v=yx8N8OCDyvc

In [32]:
Titulos_CB.to_csv('[PD]VD_Calos_50TitComent.csv')

**Pegar Todos Vid**

In [3]:
# criar função para pegar todos os vídeos de uma canal
# 1. fazer o request para o channel id
# 2. get o id dos playlistId
# 3. fazer um looping para pegar todos os vídeos do canal 
## (looping para a próxima página do yt)


def get_channel_videos(channel_id):
    
    # get Uploads playlist id
    res = youtube.channels().list(id=channel_id, 
                                  part='contentDetails').execute()
    playlist_id = res['items'][0]['contentDetails']['relatedPlaylists']['uploads']
    
    videos = []
    next_page_token = None
    
    while 1:
        res = youtube.playlistItems().list(playlistId=playlist_id, 
                                           part='snippet', 
                                           maxResults=50,
                                           pageToken=next_page_token).execute()
       
        videos += res['items']
        next_page_token = res.get('nextPageToken')
        
        if next_page_token is None:
            break
    
    return videos

In [43]:
videos = get_channel_videos('UCBrntWupyTRnnsRy3dA4VdA')


In [44]:
len(videos)

790

In [45]:
for i in videos:
    print(i["snippet"]['title'], "https://www.youtube.com/watch?v="+i['snippet']['resourceId']['videoId'])

Aeroportos de Congonhas e Navegantes passam por reformas para o bem estar do usuário. https://www.youtube.com/watch?v=oVxFFpZjigg
Ferrovia Oeste-Leste será entregue em 2022 com mais 485,4km de extensão. Detalhes na descrição: https://www.youtube.com/watch?v=W12HyI8hrvk
Duplicação da BR-101/NE está sendo feita em pavimento rígido (22 cm + 10 cm de concreto compactado) https://www.youtube.com/watch?v=EhBiTP7QDBo
De um Pernambucano ao Presidente Bolsonaro. Por isso querem tanto calar a internet, a voz do povo! https://www.youtube.com/watch?v=PDhyGCql-_0
Plano emergencial de enfrentamento à Covid19 / Operação Acolhida em Boa Vista (RR). https://www.youtube.com/watch?v=yx8N8OCDyvc
Quando esquecem de combinar com a entrevistada. https://www.youtube.com/watch?v=jVbrSEP7NKI
Fase de conclusão do Complexo Viário em Blumenau e mais BRs. Detalhes de mais obras na descrição: https://www.youtube.com/watch?v=BFx1Saet9dA
Governo do Brasil retorna dragagem do Rio Madeira. Mais detalhes na descrição: ht

Bolsonaro, Lula e a Democracia Cubana (24.02.2010) https://www.youtube.com/watch?v=J7zjA5ZXyGw
Bolsonaro explica porque Dilma Não Pode Ser Presidente (11/02/2010) https://www.youtube.com/watch?v=mpPtjF5U1fY
Bolsonaro expôe os crimes do terrorista Cesari Battisti https://www.youtube.com/watch?v=HvTXimwblak
BOLSONARO - Leitura do verdadeiro curriculum vitae de Dilma Rousseff https://www.youtube.com/watch?v=qxLWWjW3T_8
Bolsonaro x Sérgio Cabral (salário vergonhoso PMERJ) https://www.youtube.com/watch?v=s5SBCHwgNys
MELHOR DISCURSO DE BOLSONARO https://www.youtube.com/watch?v=SmeRhyFnmF0
BOLSONARO (PP-RJ) x Maria do Rosário (PT-RS) https://www.youtube.com/watch?v=atKHN_irOsQ
BOLSONARO EM 4 MINUTOS https://www.youtube.com/watch?v=2zTwmzpmfhc
BOLSONARO x TARSO GENRO (MIN. DA JUSTIÇA) https://www.youtube.com/watch?v=5sRod9azlX0
BOLSONARO X REAJUSTE SALARIAL 2007 (20/12/2007) https://www.youtube.com/watch?v=jVTigL5cJhc
Bolsonaro escovando Nelson Jobim https://www.youtube.com/watch?v=X0RKP4Ga97c

**Comentário**

In [65]:
for i in videos:
    print(i["snippet"]['title'],"https://www.youtube.com/watch?v="+i['snippet']['resourceId']['videoId'], i["snippet"]['description'], "\n")

Aeroportos de Congonhas e Navegantes passam por reformas para o bem estar do usuário. https://www.youtube.com/watch?v=oVxFFpZjigg  

Ferrovia Oeste-Leste será entregue em 2022 com mais 485,4km de extensão. Detalhes na descrição: https://www.youtube.com/watch?v=W12HyI8hrvk O Governo do Brasil  e o Ministério da Infraestrutura priorizam a construção da Ferrovia de Integração Oeste-Leste/Fiol! São mais de mil colaboradores para entregar o 2º trecho da  ferrovia em 2022. Com 485,4 km de extensão, o trecho 2, entre Caetité/BA e Barreiras/BA conectará a região produtora de grãos do oeste da Bahia ao porto de Ilhéus. 

Duplicação da BR-101/NE está sendo feita em pavimento rígido (22 cm + 10 cm de concreto compactado) https://www.youtube.com/watch?v=EhBiTP7QDBo ‪- Duplicação da BR-101/NE está sendo feita em pavimento rígido (22 cm + 10 cm de concreto compactado).‬

‪- Como determinado pelo Capitão Tarcísio  , Ministro da Infraestrutura , o mesmo serviço está sendo feito na BR-285/SC.‬ 

De um 

- Obrigado Jairo Cardoso Bolívia. 

Pesquisa realizada em arena de rodeio. https://www.youtube.com/watch?v=xfG6sXBSiUE - Pesquisa presidencial na arena.
- Pratápolis - MG 

GENERAL MOURÃO E OS RUMOS DO BRASIL. https://www.youtube.com/watch?v=SdEFp4D6zcY  

Tony Eduardo e Armas S.A. falam sobre liberdade. https://www.youtube.com/watch?v=HUasuWPBZC0  

- Prates: "É melhor JAIR se preparando para um Brasil melhor." https://www.youtube.com/watch?v=loG4l0AB1Vg  

ADESIVO "ESCOLA SEM PARTIDO". Garanta o seu! https://www.youtube.com/watch?v=inaaOY4_Hoo A lei é maior que a vontade de seu professor doutrinador! Denuncie-o! 

PORQUE A ESQUERDA É REVOLTADA COM BOLSONARO! https://www.youtube.com/watch?v=flkf9yllvDw  

BOLSONARO: A AMAZÔNIA É NOSSA? https://www.youtube.com/watch?v=OZfKNYqRMLE - A Amazônia é nossa?

- Respeitam os índios ou os mesmos são apenas massa de manobra para criação de futuras nações? 

BOLSONARO E OS ÍNDIOS: A VERDADE. https://www.youtube.com/watch?v=ZejTTmLkHs0 - Prefeito,

**Stats**

In [66]:
video_ids = list(map(lambda x:x ['snippet']['resourceId']['videoId'], videos))

def get_vid_stats (video_ids):
    stats = []
    for i in range(0, len(video_ids), 50):
        res = youtube.videos().list(id="," .join(video_ids[i:i+50]),
                                   part='statistics').execute()
        
        stats += res['items']
    
    return stats

In [67]:
stats = get_vid_stats(video_ids)

In [69]:
stats[0]

{'kind': 'youtube#video',
 'etag': 'ObKwCSbsXick5goX8DKG5tpa5aY',
 'id': 'oVxFFpZjigg',
 'statistics': {'viewCount': '6198',
  'likeCount': '2830',
  'dislikeCount': '6',
  'favoriteCount': '0',
  'commentCount': '158'}}

In [70]:
most_liked = sorted(stats, key=lambda x:int(x['statistics']['likeCount']), reverse=True)

In [71]:
for i in most_liked :
    print(i['statistics']['likeCount'], i['statistics']['dislikeCount'], "https://www.youtube.com/watch?v="+i['id'], "\n")

94764 1783 https://www.youtube.com/watch?v=JyN4hhTGdlQ 

72206 1770 https://www.youtube.com/watch?v=uQGbbqGVCjY 

45310 224 https://www.youtube.com/watch?v=jVbrSEP7NKI 

42683 264 https://www.youtube.com/watch?v=pxeYAX6issU 

42456 1413 https://www.youtube.com/watch?v=CzyxR6HvWkg 

28695 532 https://www.youtube.com/watch?v=pcSm9adaufc 

28528 579 https://www.youtube.com/watch?v=7QTk-Sctiw0 

28429 581 https://www.youtube.com/watch?v=Dv3WPK7V7Zo 

26970 152 https://www.youtube.com/watch?v=UJDm7h2UdnE 

25607 103 https://www.youtube.com/watch?v=fZzmIQvjcZo 

24921 288 https://www.youtube.com/watch?v=4wMQ2eMraig 

24751 185 https://www.youtube.com/watch?v=X0XqXOpPpsE 

23981 201 https://www.youtube.com/watch?v=Is4YFBB8C64 

23857 135 https://www.youtube.com/watch?v=qwV74sWhp4s 

23799 483 https://www.youtube.com/watch?v=pmb523Zx-5Q 

23785 121 https://www.youtube.com/watch?v=SPNU0lkesUY 

22574 211 https://www.youtube.com/watch?v=UYkfvp85ohk 

22511 61 https://www.youtube.com/watch?v=PDhy

1173 0 https://www.youtube.com/watch?v=HvjXqk7hJ_0 

1165 8 https://www.youtube.com/watch?v=vZSXdKkXC74 

1161 5 https://www.youtube.com/watch?v=2__e1WfGzKg 

1159 2 https://www.youtube.com/watch?v=vqn1Yvi9jOQ 

1142 2 https://www.youtube.com/watch?v=C2uWMs7BIvw 

1138 86 https://www.youtube.com/watch?v=aFQhCmRcOt0 

1130 3 https://www.youtube.com/watch?v=IrqbmWpnPCA 

1126 1 https://www.youtube.com/watch?v=-rVuk3fv7Uk 

1125 11 https://www.youtube.com/watch?v=CSMLG3crupc 

1112 22 https://www.youtube.com/watch?v=yVLI5QUZaso 

1100 4 https://www.youtube.com/watch?v=7DD6Ynr2y10 

1099 204 https://www.youtube.com/watch?v=NsD1J02cawY 

1097 219 https://www.youtube.com/watch?v=Dew5K_4BXbU 

1096 4 https://www.youtube.com/watch?v=0smQkl57N9o 

1090 3 https://www.youtube.com/watch?v=fWmgZZzDnVk 

1087 7 https://www.youtube.com/watch?v=GdTiBFOQqmo 

1045 0 https://www.youtube.com/watch?v=Facd4_5B5SY 

1042 2 https://www.youtube.com/watch?v=GgrLK1Jv9gI 

1035 12 https://www.youtube.com/watch?v

In [77]:
most_comment = sorted(stats, key=lambda x:int(x['statistics']['commentCount']), reverse=True)

for i in most_liked:
    print(i['statistics']['commentCount'], i['statistics']['likeCount'], i['statistics']['dislikeCount'], "https://www.youtube.com/watch?v="+i['id'], "\n")

6403 94764 1783 https://www.youtube.com/watch?v=JyN4hhTGdlQ 

4725 72206 1770 https://www.youtube.com/watch?v=uQGbbqGVCjY 

3800 45310 224 https://www.youtube.com/watch?v=jVbrSEP7NKI 

4247 42683 264 https://www.youtube.com/watch?v=pxeYAX6issU 

8202 42456 1413 https://www.youtube.com/watch?v=CzyxR6HvWkg 

2706 28695 532 https://www.youtube.com/watch?v=pcSm9adaufc 

2898 28528 579 https://www.youtube.com/watch?v=7QTk-Sctiw0 

2434 28429 581 https://www.youtube.com/watch?v=Dv3WPK7V7Zo 

2028 26970 152 https://www.youtube.com/watch?v=UJDm7h2UdnE 

2831 25607 103 https://www.youtube.com/watch?v=fZzmIQvjcZo 

2646 24921 288 https://www.youtube.com/watch?v=4wMQ2eMraig 

2630 24751 185 https://www.youtube.com/watch?v=X0XqXOpPpsE 

1665 23981 201 https://www.youtube.com/watch?v=Is4YFBB8C64 

1774 23857 135 https://www.youtube.com/watch?v=qwV74sWhp4s 

3826 23799 483 https://www.youtube.com/watch?v=pmb523Zx-5Q 

2612 23785 121 https://www.youtube.com/watch?v=SPNU0lkesUY 

2273 22574 211 https:

86 1654 2 https://www.youtube.com/watch?v=KMM1APUEcfE 

187 1636 7 https://www.youtube.com/watch?v=v-mjAYaIiTE 

165 1587 0 https://www.youtube.com/watch?v=LQhtSEA_rOw 

170 1577 7 https://www.youtube.com/watch?v=eCmqSCPbRrQ 

283 1577 4 https://www.youtube.com/watch?v=h2QmYaBHJFc 

236 1568 4 https://www.youtube.com/watch?v=2nnpp2aLJJs 

140 1554 6 https://www.youtube.com/watch?v=BFx1Saet9dA 

192 1554 13 https://www.youtube.com/watch?v=8Jz3EGukSI0 

155 1549 9 https://www.youtube.com/watch?v=cmUvJ4g04Mw 

80 1538 2 https://www.youtube.com/watch?v=SNJtA3JQXcs 

226 1513 4 https://www.youtube.com/watch?v=-8JQUHQ42tk 

304 1494 4 https://www.youtube.com/watch?v=Z6ZdLU2GRFk 

129 1493 2 https://www.youtube.com/watch?v=poFjvggl1AI 

148 1492 8 https://www.youtube.com/watch?v=_cbHI6EG-j4 

68 1479 32 https://www.youtube.com/watch?v=o-47amg9VXM 

186 1476 3 https://www.youtube.com/watch?v=Hr-KyCKtct8 

113 1475 1 https://www.youtube.com/watch?v=hzkONa_DaJ8 

92 1473 4 https://www.youtube.co

17 245 1 https://www.youtube.com/watch?v=i0vCwRYNeq4 

17 245 18 https://www.youtube.com/watch?v=ZejTTmLkHs0 

56 245 27 https://www.youtube.com/watch?v=6zP7R6xPYEk 

13 243 2 https://www.youtube.com/watch?v=GHRQe-VQxMI 

13 243 2 https://www.youtube.com/watch?v=_eCgf0R2Wx8 

15 243 0 https://www.youtube.com/watch?v=WGky3zOq0pQ 

10 242 8 https://www.youtube.com/watch?v=bWBHGC2AxPU 

20 242 3 https://www.youtube.com/watch?v=OmGxaWGAgyE 

28 233 2 https://www.youtube.com/watch?v=RmCIfiF9TvQ 

59 232 4 https://www.youtube.com/watch?v=5KisVXYACxs 

5 225 12 https://www.youtube.com/watch?v=xIyn-P-H3iU 

28 223 2 https://www.youtube.com/watch?v=rxNzoLNNcbo 

15 220 0 https://www.youtube.com/watch?v=ot-44PorwY4 

20 219 1 https://www.youtube.com/watch?v=YPWlMYrXuQ0 

5 218 2 https://www.youtube.com/watch?v=eMMaSWx2m8U 

11 217 0 https://www.youtube.com/watch?v=lXN6nfb404M 

14 216 2 https://www.youtube.com/watch?v=9cHpZ-TUEOY 

8 214 4 https://www.youtube.com/watch?v=X9AVOn77d2k 

11 213 1 h

In [86]:
Stts02 = []

for i in most_liked:
    comment =  i['statistics']['commentCount']
    view = i['statistics']['viewCount']
    favorite = i['statistics']['favoriteCount']
    like = i['statistics']['likeCount'] 
    dislike = i['statistics']['dislikeCount'] 
    link = ("https://www.youtube.com/watch?v="+i['id'])
    Stts02.append((comment, view, favorite, like, dislike, link))
    

Carlos_Stats1 = pd.DataFrame(Stts02, columns=['comment', 'view', 'favorite', 'like', 'dislike',  'link'])

Carlos_Stats1.head(6)

comment     view favorite   like dislike  \
0    6403  1623694        0  94764    1783   
1    4725  1455495        0  72206    1770   
2    3800   202761        0  45310     224   
3    4247   210372        0  42683     264   
4    8202  1847141        0  42456    1413   
5    2706   230133        0  28695     532   

                                          link  
0  https://www.youtube.com/watch?v=JyN4hhTGdlQ  
1  https://www.youtube.com/watch?v=uQGbbqGVCjY  
2  https://www.youtube.com/watch?v=jVbrSEP7NKI  
3  https://www.youtube.com/watch?v=pxeYAX6issU  
4  https://www.youtube.com/watch?v=CzyxR6HvWkg  
5  https://www.youtube.com/watch?v=pcSm9adaufc

In [87]:
Carlos_Stats1.to_csv('[PD]Carlos_Stats1.csv')

<h1>Eduardo</h1>

In [4]:
#UCkR6xPOHhpjq3wnFchVI4sg

q_edub =youtube.channels().list(id = "UCkR6xPOHhpjq3wnFchVI4sg", part = "contentDetails")
canal_edub= q_edub.execute()

In [5]:
canal_edub

{'kind': 'youtube#channelListResponse',
 'etag': '91bvfZ5mbskELdK94bbfc2WXpaE',
 'pageInfo': {'totalResults': 1, 'resultsPerPage': 1},
 'items': [{'kind': 'youtube#channel',
   'etag': '5e_DZ4xqZsrhbJKdgUmig4hRQtM',
   'id': 'UCkR6xPOHhpjq3wnFchVI4sg',
   'contentDetails': {'relatedPlaylists': {'likes': '',
     'favorites': '',
     'uploads': 'UUkR6xPOHhpjq3wnFchVI4sg',
     'watchHistory': 'HL',
     'watchLater': 'WL'}}}]}

In [104]:
# Fazer requerimento de lista de vídeos - os 50
#'uploads': 'UUkR6xPOHhpjq3wnFchVI4sg'

content = youtube.playlistItems().list(playlistId = "UUkR6xPOHhpjq3wnFchVI4sg", part = "snippet", 
                                       maxResults = 50)

Edu_playlist = content.execute()

In [123]:
Edu_playlist['items'][0]

{'kind': 'youtube#playlistItem',
 'etag': 'YfB82UDkZ5PrW4sB2145SrGoBRc',
 'id': 'VVVrUjZ4UE9IaHBqcTN3bkZjaFZJNHNnLlFEM0dXOEhFNUhJ',
 'snippet': {'publishedAt': '2020-07-22T18:31:04Z',
  'channelId': 'UCkR6xPOHhpjq3wnFchVI4sg',
  'title': 'Paulo Guedes defende reforma tributária ao lado de Maia e Alcolumbre',
  'description': '',
  'thumbnails': {'default': {'url': 'https://i.ytimg.com/vi/QD3GW8HE5HI/default.jpg',
    'width': 120,
    'height': 90},
   'medium': {'url': 'https://i.ytimg.com/vi/QD3GW8HE5HI/mqdefault.jpg',
    'width': 320,
    'height': 180},
   'high': {'url': 'https://i.ytimg.com/vi/QD3GW8HE5HI/hqdefault.jpg',
    'width': 480,
    'height': 360},
   'standard': {'url': 'https://i.ytimg.com/vi/QD3GW8HE5HI/sddefault.jpg',
    'width': 640,
    'height': 480},
   'maxres': {'url': 'https://i.ytimg.com/vi/QD3GW8HE5HI/maxresdefault.jpg',
    'width': 1280,
    'height': 720}},
  'channelTitle': 'EDUARDO BOLSONARO',
  'playlistId': 'UUkR6xPOHhpjq3wnFchVI4sg',
  'position':

**50 Vids**

In [109]:
# pegar 50 vídeo = ['title'][link]    

for r in Edu_playlist['items']:
    print(r['snippet']['title'], r['snippet']['description'],"https://www.youtube.com/watch?v="+r['snippet']['resourceId']['videoId'], "\n")

Paulo Guedes defende reforma tributária ao lado de Maia e Alcolumbre  https://www.youtube.com/watch?v=QD3GW8HE5HI 

9º manifestação pró Bolsonaro (19/JUL/2020) E a luta não se limita a uma pessoa ou a um governo, é pela liberdade, pelas futuras gerações🇧🇷🙏 https://www.youtube.com/watch?v=nb2OJzIri0o 

O BRASIL PRECISA SABER: Gilson Machado O Deputado Federal Eduardo Bolsonaro recebe Gilson Machado Neto, Presidente da Embratur.

Na pauta, o futuro do turismo no Brasil e muito mais. Confira! https://www.youtube.com/watch?v=1a2n0wSbBno 

Eduardo Bolsonaro ao vivo no Pampa Debates  https://www.youtube.com/watch?v=SnVUvW94ghk 

Fox News: Bolsonaro, Trump e cloroquina (JUL/2020) https://www.youtube.com/watch?v=dbR5E3ovlRk 

Venezuelano Rafael Valera responde Nelipe Feto  https://www.youtube.com/watch?v=TVC_D16kN9E 

Deputados falam em defesa das liberdades e respeito à constituição Manifestação ocorrida na tarde de 15/JUL/2020. Neste vídeo falam os deputados federais:

-Gen. Girão
-Bia Kicis

**Salvar**

In [122]:
Titulos_Edu = []

for r in Edu_playlist['items']:
    Title = r['snippet']['title'] 
    Description = r['snippet']['description']
    Link = "https://www.youtube.com/watch?v="+r['snippet']['resourceId']['videoId']
    Titulos_Edu.append((Title,Description,Link))
    
Titulos_Edu = pd.DataFrame(Titulos_Edu, columns = ['Title','Description','Link'])
    

Titulos_Edu.to_csv('[PD]VD_Edu_50TitComent.csv')

#Titulos_Edu.head()

**Pegar Todos Vid**

In [6]:
# criar função para pegar todos os vídeos de uma canal
# 1. fazer o request para o channel id
# 2. get o id dos playlistId
# 3. fazer um looping para pegar todos os vídeos do canal 
## (looping para a próxima página do yt)


def get_channel_videos(channel_id):
    
    # get Uploads playlist id
    res = youtube.channels().list(id=channel_id, 
                                  part='contentDetails').execute()
    playlist_id = res['items'][0]['contentDetails']['relatedPlaylists']['uploads']
    
    videos = []
    next_page_token = None
    
    while 1:
        res = youtube.playlistItems().list(playlistId=playlist_id, 
                                           part='snippet', 
                                           maxResults=50,
                                           pageToken=next_page_token).execute()
       
        videos += res['items']
        next_page_token = res.get('nextPageToken')
        
        if next_page_token is None:
            break
    
    return videos

In [7]:
videos = get_channel_videos('UCkR6xPOHhpjq3wnFchVI4sg')



In [12]:
videos

[{'kind': 'youtube#playlistItem',
  'etag': 'FObnOGGl5DbZ3m2TjSy1HVORxVo',
  'id': 'VVVrUjZ4UE9IaHBqcTN3bkZjaFZJNHNnLjJ5aldrTDlXR1ZV',
  'snippet': {'publishedAt': '2020-07-30T10:35:04Z',
   'channelId': 'UCkR6xPOHhpjq3wnFchVI4sg',
   'title': 'Netinho e a entrevista mais polêmica de sua vida - assista no próximo sábado (teaser)',
   'description': 'Neste sábado, Eduardo Bolsonaro recebe o cantor Netinho no programa O Brasil Precisa Saber\u202c.\n\n\u202aNa entrevista mais reveladora de sua vida, ele fala sobre conservadorismo, homossexualismo e movimento LGBT, Globo, sexo, suicídio, carnaval, política e muito mais\u202c.\n\n\u202aConfira aqui, no YouTube, neste sábado (01 de agosto), às 18h.',
   'thumbnails': {'default': {'url': 'https://i.ytimg.com/vi/2yjWkL9WGVU/default.jpg',
     'width': 120,
     'height': 90},
    'medium': {'url': 'https://i.ytimg.com/vi/2yjWkL9WGVU/mqdefault.jpg',
     'width': 320,
     'height': 180},
    'high': {'url': 'https://i.ytimg.com/vi/2yjWkL9WGVU/

In [8]:
len(videos)

1386

In [17]:
for i in videos:
    print(i["snippet"]['channelTitle'], i["snippet"]['title'],"https://www.youtube.com/watch?v="+i['snippet']['resourceId']['videoId'], i["snippet"]['description'], "\n")

EDUARDO BOLSONARO Netinho e a entrevista mais polêmica de sua vida - assista no próximo sábado (teaser) https://www.youtube.com/watch?v=2yjWkL9WGVU Neste sábado, Eduardo Bolsonaro recebe o cantor Netinho no programa O Brasil Precisa Saber‬.

‪Na entrevista mais reveladora de sua vida, ele fala sobre conservadorismo, homossexualismo e movimento LGBT, Globo, sexo, suicídio, carnaval, política e muito mais‬.

‪Confira aqui, no YouTube, neste sábado (01 de agosto), às 18h. 

EDUARDO BOLSONARO Bolsonaro veste a camisa do Corinthians com Marcelinho Carioca! https://www.youtube.com/watch?v=4-0SaTclpIk Mesmo palmeirense, hoje o Presidente Jair Bolsonaro vestiu a camisa do Corinthians, para defender a MP que trouxe liberdade aos clubes de futebol.‬

‪Ao lado de Marcelinho Carioca, o Presidente mostrou que essa é uma luta de todas as torcidas contra monopólio de transmissão global. 

EDUARDO BOLSONARO Desencarceramento, base da criminalidade https://www.youtube.com/watch?v=ntwubMpc_KA Quem nunca

EDUARDO BOLSONARO Foro de São Paulo aterroriza América Latina https://www.youtube.com/watch?v=ZLnPr1hA-Vg  

EDUARDO BOLSONARO Sessão Solene Dia Nacional de Valorização da Família https://www.youtube.com/watch?v=PY14WNJQUcQ  

EDUARDO BOLSONARO Live sobre o que tem ocorrido com o PSL (19/OUT/2019) https://www.youtube.com/watch?v=SNtooQsoLEQ  

EDUARDO BOLSONARO Entenda o que ocorre com o PSL (Folha do Brasil) https://www.youtube.com/watch?v=iXXBG2DBS5c  

EDUARDO BOLSONARO Ministro Arthur Weintraub no CPAC 2019 https://www.youtube.com/watch?v=4ZJavgrhwQc Evento ocorrido em 11 e 12/OUT no Hotel Transamerica em São Paulo. 

EDUARDO BOLSONARO Conselho do Olavo para mimizentos https://www.youtube.com/watch?v=702x24UG5_8 #farmeme 

EDUARDO BOLSONARO Dinheiro público no CPAC https://www.youtube.com/watch?v=cuySBD3hwTM "Aiiiiin gastaram dinheiro público no CPAC". Depois chamam de isentão e eles reclamam. Assista e veja o que de fato ocorre.

Divulgue os vídeos do @cpacbrasil👇
-11/OUT:
yo

EDUARDO BOLSONARO Eduardo Bolsonaro adota um vira lata (21/ABR/2018) https://www.youtube.com/watch?v=flFh2X2jtG4 (21/ABR/2018) ADOTAR OU NÃO ADOTAR UM ANIMAL?  Agradeço aos instagrams de cabanadopicapau (goo.gl/Pkf2T6) e lartemporariooasis (goo.gl/YsHntS) pela iniciativa voluntária de intermediar essas adoções que levam carinho e saúde aos animais 👏👏👏 

EDUARDO BOLSONARO Mulheres se emocionam ao ver Bolsonaro (ABR/2017) https://www.youtube.com/watch?v=v0dQlOheDVI Seja no gabinete ou num shopping, a cada demonstração de carinho, o crescimento da responsabilidade. 

EDUARDO BOLSONARO VIOLÊNCIA: É PRECISO ENCARAR DE FRENTE. https://www.youtube.com/watch?v=ZbBC3KWGxvY POR QUE O PACÍFICO BRASILEIRO SE TORNOU UM POVO TÃO VIOLENTO?

A evolução legislativa após a constituição de 1988 demonstra cabalmente a luta pelos ditos defensores dos direitos humanos em prol de garantias jurídicas para criminosos. O exagero foi de tal forma avassalador que o bandido virou vítima e a vítima o bandido. Assis

EDUARDO BOLSONARO “Abraham Lincoln tornou todos os homens livres, mas Samuel Colt os tornou iguais” https://www.youtube.com/watch?v=5CU06yyH-iY Brasileiro residente nos EUA mostra suas armas de casa - com Yves da 88 Tactical , Tony Eduardo do Clube e Escola de Tiro .38 e Carlos Bolsonaro.

“Eu comprei as armas originais, armas de verdade iguais às que eu nós usamos no filme Tomb Raider para segurança. Brad e eu não somos contra a posse de armas em casa, e até temos uma. E sim, nós estamos dispostos a usá-la se for necessário… Se alguém entrar na minha casa e tentar ferir minhas crianças, eu não vejo nenhum problema em atirar.” ( Angelina Jolie ) http://spotniks.com/22-celebridades-que-sao-contra-o-desarmamento-civil/ 

EDUARDO BOLSONARO QUE PAÍS É ESSE? https://www.youtube.com/watch?v=ZcMbd5YDWXg Algo está muito errado! 

EDUARDO BOLSONARO Rifa para ensinar crianças a atirar https://www.youtube.com/watch?v=iOaXsSiex98 (19/JAN/2017) TODO CIDADÃO DEVE SER UM SOLDADO
 "Todo cidadão deve s

EDUARDO BOLSONARO Fazendeiro do Mato Grosso do Sul pedem socorro contra invasões indígenas https://www.youtube.com/watch?v=5KReywj2gxI Fazendeira Luana faz apelo para evitar conflitos com indígenas no Mato Grosso do Sul. A imprensa não cobre. O governo fica inerte. Os agricultores foram expulsos de suas casas e seguem desamparados. Compartilhe! 

EDUARDO BOLSONARO PARTE 1 -  Dep. Jair Bolsonaro para Rádio Pampa https://www.youtube.com/watch?v=CXUO0e-Fa9s Deputado Jair Bolsonaro conversa sobre os mais variados temas de interesse do Brasil. 

EDUARDO BOLSONARO Bolsonaro na Av. Paulista! https://www.youtube.com/watch?v=MTyaYJCue3Q Durante as manifestações do dia 16 de agosto, o Dep. Eduardo Bolsonaro, discursou para um público que não suporta mais tanta roubalheira no país! 

EDUARDO BOLSONARO DEP. JAIR BOLSONARO EM FORTALEZA https://www.youtube.com/watch?v=HeI9mKoBY4A Entrevista concedida no Centro Industrial do Ceará em agosto de 2015. 

EDUARDO BOLSONARO SÃO PAULO DIZ: FORA PT! https:/

In [21]:
Vids_Edu = []

for i in videos:
    ChannelTitle = i["snippet"]['channelTitle']
    PublishedAt = i["snippet"]['publishedAt']
    Title = i["snippet"]['title']
    Description = i["snippet"]['description'] 
    Id = i['snippet']['resourceId']['videoId']
    Link = "https://www.youtube.com/watch?v=" +i['snippet']['resourceId']['videoId']
    Vids_Edu.append((ChannelTitle, PublishedAt, Title, Description, Id, Link))
    
Vids_Edu = pd.DataFrame(Vids_Edu, columns = ['ChannelTitle', 'PublishedAt','Title', 'Description', 'Id', 'Link'])

Vids_Edu.PublishedAt = pd.to_datetime(Vids_Edu.PublishedAt ) 

Vids_Edu.head(5)

Vids_Edu.to_csv('[PD]VD_EduB_30072020.csv')

**Stats**

In [35]:
video_ids = list(map(lambda x:x ['snippet']['resourceId']['videoId'], videos))

def get_vid_stats (video_ids):
    
    stats = []
    
    for i in range(0, len(video_ids), 50):
        res = youtube.videos().list(id=",".join(video_ids[i:i+50]),
                                   part='statistics').execute()
        
        stats += res['items']
    
    return stats

In [36]:
stats = get_vid_stats(video_ids)

In [24]:
stats_liked = sorted(stats, key=lambda x:int(x['statistics']['likeCount']), reverse=True)

In [37]:
stats

[{'kind': 'youtube#video',
  'etag': 'f3sjVIDpYq2OqKx2U7y6PpRTIiA',
  'id': '2yjWkL9WGVU',
  'statistics': {'viewCount': '38373',
   'likeCount': '13097',
   'dislikeCount': '66',
   'favoriteCount': '0',
   'commentCount': '890'}},
 {'kind': 'youtube#video',
  'etag': 'PMUGY8_VWf66V-vd4TJrDU6r5eI',
  'id': '4-0SaTclpIk',
  'statistics': {'viewCount': '60435',
   'likeCount': '16355',
   'dislikeCount': '307',
   'favoriteCount': '0',
   'commentCount': '1630'}},
 {'kind': 'youtube#video',
  'etag': 'iuE9cvkyPf-5ToLyO7a_uZYUYfM',
  'id': 'ntwubMpc_KA',
  'statistics': {'viewCount': '14213',
   'likeCount': '4635',
   'dislikeCount': '15',
   'favoriteCount': '0',
   'commentCount': '497'}},
 {'kind': 'youtube#video',
  'etag': 'GojapZg5siRxfJAhE6OJ3gyOjLI',
  'id': 'jRYWw76wobs',
  'statistics': {'viewCount': '13180',
   'likeCount': '5393',
   'dislikeCount': '17',
   'favoriteCount': '0',
   'commentCount': '427'}},
 {'kind': 'youtube#video',
  'etag': 'YFt71E70gapTZozs3cudHqUneJQ',


In [38]:
for i in stats:
    print(i['statistics']['likeCount'], 
          i['statistics']['dislikeCount'],
          i['statistics']['commentCount'],
          i['id'],
          "https://www.youtube.com/watch?v="+i['id'])

13097 66 890 2yjWkL9WGVU https://www.youtube.com/watch?v=2yjWkL9WGVU
16355 307 1630 4-0SaTclpIk https://www.youtube.com/watch?v=4-0SaTclpIk
4635 15 497 ntwubMpc_KA https://www.youtube.com/watch?v=ntwubMpc_KA
5393 17 427 jRYWw76wobs https://www.youtube.com/watch?v=jRYWw76wobs
32627 354 2251 VYFY4FASyRg https://www.youtube.com/watch?v=VYFY4FASyRg
5478 13 425 VMqubSpjkYE https://www.youtube.com/watch?v=VMqubSpjkYE
28962 240 2496 tVPbCSFM59Y https://www.youtube.com/watch?v=tVPbCSFM59Y
3875 21 406 QD3GW8HE5HI https://www.youtube.com/watch?v=QD3GW8HE5HI
10545 80 933 nb2OJzIri0o https://www.youtube.com/watch?v=nb2OJzIri0o
7528 38 603 1a2n0wSbBno https://www.youtube.com/watch?v=1a2n0wSbBno
14481 123 976 SnVUvW94ghk https://www.youtube.com/watch?v=SnVUvW94ghk
3636 12 227 dbR5E3ovlRk https://www.youtube.com/watch?v=dbR5E3ovlRk
14648 89 1176 TVC_D16kN9E https://www.youtube.com/watch?v=TVC_D16kN9E
6121 90 671 rwSCh_zQPBI https://www.youtube.com/watch?v=rwSCh_zQPBI
7493 23 619 pVGyOjBf4AE https://w

KeyError: 'commentCount'

In [40]:
stats_eb = []

for i in stats:
    view = i['statistics']['viewCount']
    favorite = i['statistics']['favoriteCount']
    like = i['statistics']['likeCount'] 
    dislike = i['statistics']['dislikeCount']
    #commentCount = i['statistics']['commentCount']
    Id = i['id']
    link = "https://www.youtube.com/watch?v="+i['id']
    stats_eb.append((view, favorite, like, dislike, Id, link))
    

stats_eb = pd.DataFrame(stats_eb, columns = ['View','Favorite', 'Like', 'Dislike', "Id", 'Link'])

stats_eb

In [43]:
stats_eb.to_csv('[PD]Edu_Stats2_3007202.csv')

In [41]:
stats_eb.head(5)

View Favorite   Like Dislike           Id  \
0   38373        0  13097      66  2yjWkL9WGVU   
1   60435        0  16355     307  4-0SaTclpIk   
2   14213        0   4635      15  ntwubMpc_KA   
3   13180        0   5393      17  jRYWw76wobs   
4  114077        0  32627     354  VYFY4FASyRg   

                                          Link  
0  https://www.youtube.com/watch?v=2yjWkL9WGVU  
1  https://www.youtube.com/watch?v=4-0SaTclpIk  
2  https://www.youtube.com/watch?v=ntwubMpc_KA  
3  https://www.youtube.com/watch?v=jRYWw76wobs  
4  https://www.youtube.com/watch?v=VYFY4FASyRg

<h1>Jair</h1>

In [44]:
#UC8hGUtfEgvvnp6IaHSAg1OQ
q_Jair =youtube.channels().list(id = "UC8hGUtfEgvvnp6IaHSAg1OQ", part = "contentDetails")
canal_Jair = q_Jair.execute()

In [184]:
canal_Jair['items'][0]

{'kind': 'youtube#channel',
 'etag': '3LvrLTygku2hPINQc2A1VGHuqB8',
 'id': 'UC8hGUtfEgvvnp6IaHSAg1OQ',
 'contentDetails': {'relatedPlaylists': {'likes': '',
   'favorites': '',
   'uploads': 'UU8hGUtfEgvvnp6IaHSAg1OQ',
   'watchHistory': 'HL',
   'watchLater': 'WL'}}}

In [174]:
#uploads': 'UC8hGUtfEgvvnp6IaHSAg1OQ'
#id: 'UU8hGUtfEgvvnp6IaHSAg1OQ'

In [186]:
content_J = youtube.playlistItems().list(playlistId = "UU8hGUtfEgvvnp6IaHSAg1OQ", part = "snippet", 
                                       maxResults = 50)

Jair = content_J.execute()

In [187]:
Jair['items'][0]

{'kind': 'youtube#playlistItem',
 'etag': 'lvsrDwR2GX6OkYBLc2GOStvAONs',
 'id': 'VVU4aEdVdGZFZ3Z2bnA2SWFIU0FnMU9RLmZJYTJaUnlfYV9z',
 'snippet': {'publishedAt': '2020-07-22T10:17:56Z',
  'channelId': 'UC8hGUtfEgvvnp6IaHSAg1OQ',
  'title': 'Reforço do Governo do Brasil no Rio Grande do Sul com ventiladores pulmonares  e monitores cardíacos',
  'description': 'O ministro Pazuello esteve hoje na capital gaúcha. Com o governador do estado e o prefeito de Porto Alegre, anunciou a entrega de mais ventiladores pulmonares e monitores cardíacos para o combate ao coronavírus.\nÉ o Governo do Brasil trabalhando lado a lado com os estados na luta contra a #Covid19. \n\nSaiba mais 👉🏻 https://bit.ly/3juGidx',
  'thumbnails': {'default': {'url': 'https://i.ytimg.com/vi/fIa2ZRy_a_s/default.jpg',
    'width': 120,
    'height': 90},
   'medium': {'url': 'https://i.ytimg.com/vi/fIa2ZRy_a_s/mqdefault.jpg',
    'width': 320,
    'height': 180},
   'high': {'url': 'https://i.ytimg.com/vi/fIa2ZRy_a_s/hqdefau

**50 Vids**

In [188]:
for r in Jair['items']:
     print(r['snippet']['title'], r['snippet']['description'],"https://www.youtube.com/watch?v="+r['snippet']['resourceId']['videoId'], "\n")

Reforço do Governo do Brasil no Rio Grande do Sul com ventiladores pulmonares  e monitores cardíacos O ministro Pazuello esteve hoje na capital gaúcha. Com o governador do estado e o prefeito de Porto Alegre, anunciou a entrega de mais ventiladores pulmonares e monitores cardíacos para o combate ao coronavírus.
É o Governo do Brasil trabalhando lado a lado com os estados na luta contra a #Covid19. 

Saiba mais 👉🏻 https://bit.ly/3juGidx https://www.youtube.com/watch?v=fIa2ZRy_a_s 

- C130 da Força Aérea Brasileira (FAB) no combate ao fogo na Serra da Mantiqueira.  https://www.youtube.com/watch?v=kI7HAFxe_k0 

Foco nas variações de modais de transporte: Governo Bolsonaro avança no setor aquaviário.  https://www.youtube.com/watch?v=qYaABSqlzck 

- O AGRO é Brasil, é a locomotiva da nossa Economia. - Parabéns Pernambuco.  https://www.youtube.com/watch?v=QbwyqWh0LSg 

O brasileiro só quer colocar o alimento na mesa de sua família!  https://www.youtube.com/watch?v=jD11ENsWDck 

- Complicou? 

**Salvar**

In [192]:
Titulos_Jair = []

for r in Jair['items']:
    Title = r['snippet']['title'] 
    Description = r['snippet']['description']
    Link = "https://www.youtube.com/watch?v="+r['snippet']['resourceId']['videoId']
    Titulos_Jair.append((Title,Description,Link))
    
Titulos_Jair = pd.DataFrame(Titulos_Jair, columns = ['Title','Description','Link'])
    

Titulos_Jair.to_csv('[PD]VD_Jair_50TitComent.csv')

#Titulos_Jair.head(10)

**Pegar Todos Vid**

criar função para pegar todos os vídeos de uma canal
1. fazer o request para o channel id
2. get o id dos playlistId
3. fazer um looping para pegar todos os vídeos do canal (looping para a próxima página do yt)


def get_channel_videos(channel_id):
    
    # get Uploads playlist id
    res = youtube.channels().list(id=channel_id, 
                                  part='contentDetails').execute()
    playlist_id = res['items'][0]['contentDetails']['relatedPlaylists']['uploads']
    
    videos = []
    next_page_token = None
    
    while 1:
        res = youtube.playlistItems().list(playlistId=playlist_id, 
                                           part='snippet', 
                                           maxResults=50,
                                           pageToken=next_page_token).execute()
       
        videos += res['items']
        next_page_token = res.get('nextPageToken')
        
        if next_page_token is None:
            break
    
    return videos

In [45]:
videosJ = get_channel_videos('UC8hGUtfEgvvnp6IaHSAg1OQ')
len(videosJ)

2177

In [46]:
for i in videosJ:
    print(i["snippet"]['title'],"https://www.youtube.com/watch?v="+i['snippet']['resourceId']['videoId'], i["snippet"]['description'], "\n")

Presidente Jair Bolsonaro em Campo Alegre de Lourdes/BA. 30/07/2020). https://www.youtube.com/watch?v=BPUyiUUpEcg  

- São Raimundo Nonato/PI. - Aeroporto. (30/07/2020) https://www.youtube.com/watch?v=lIQMZRY_IDs  

Ministro Paulo Guedes esclarece alguns pontos da reforma tributária. Mais etalhes nas redes sociais. https://www.youtube.com/watch?v=4lGv7PCgbW4  

- Marcelinho Carioca e a MP do futebol. - No final opine quem, de fato, é melhor. https://www.youtube.com/watch?v=Z7c-gbhKmSk  

- Ampliação do aeroporto de Angra dos Reis/RJ. - Potencialização do turismo. https://www.youtube.com/watch?v=lt2IIg6p_MY  

Edifício Joelma/SP, 1974. Sgt CASSANIGA salta de helicóptero no terraço do edifício p/ salvar vidas https://www.youtube.com/watch?v=5f2wY39r7RI -Edifício Joelma/SP, 1974.

-Sgt CASSANIGA salta de helicóptero da FAB no terraço do edifício em chamas para salvar vidas, em uma das maiores tragédias na história do Brasil. 

- Água para quem tem sede. - Liberdade para um povo.  - Brasil

Entenda a MP do Agronegócio assinada pelo Presidente BOLSONARO. https://www.youtube.com/watch?v=5joktbRwc58 MP do Agro

Assinamos a Medida Provisória 897, que é um divisor de águas para o crédito rural brasileiro. A MP, que foi publicada hoje, também traz várias ferramentas que vão simplificar e dar segurança para quem quer investir no Brasil. A Medida vai possibilitar a alocação de R$ 5 bilhões a mais de crédito rural para o setor. A MP complementa medidas previstas no Plano Safra 2019/2020. 
Agradecimento especial ao nosso presidente Jair Messias BOLSONARO, ao ministro Paulo Guedes, ao ministro Roberto Campos e a toda equipe do Mapa. 

O ocidente perdeu a fé na masculinidade. Via Canal Tradutores de Direita. https://www.youtube.com/watch?v=CzLMlA4C5NY Jordan Peterson, em entrevista, é questionado sobre o porquê dos homens estarem em crise.

A resposta de Peterson é um ensaio sobre a destruição do ideal da masculinidade e a resposta necessária a esse atentado: como o homem deve enxerg

Mais sobre o intenso trabalho em Brumadinho-MG. Que Deus conforte todos! https://www.youtube.com/watch?v=_-UB-iHQGEI  

Presidente com quadro de pneumonia: não caiam em sensacionalismo. A verdade: https://www.youtube.com/watch?v=DCo3_o-jvhA  

Fomentar o pequeno comércio, geração de empregos e e crescimento econômico local. https://www.youtube.com/watch?v=0WsaLm9w-4g 🍻| As cervejas artesanais ganharam o público. O número de cervejarias autorizadas cresceu 23% no ano passado. Agora, uma novidade vai impulsionar ainda mais o setor. O registro de novas cervejas será todo digital. Detalhes com Pablo Mundim da @tvnbr 

A terceira cirurgia foi bem mais delicada que a segunda, mas o Presidente vai voltando! https://www.youtube.com/watch?v=L2cQbLYdEzk ‪Bandidos torcem para sua morte, acusando os outros do que eles são! As orações são mais fortes, e este homem não está aqui em vão. A terceira cirurgia foi bem mais delicada que a segunda, mas o Presidente escolhido pelo povo vai voltando! A cara

O desespero dos sanguessugas da política: Mentem, distorcem ou potencializam tudo sobre Bolsonaro. https://www.youtube.com/watch?v=-EhHUYSXF-o  

- O Brasil não suporta mais 4 anos de PT, PSDB e aliados. Eis  porque todos estão contra nós. https://www.youtube.com/watch?v=ewmL6WrozaM  

O Datafolha e o vexame de sempre. https://www.youtube.com/watch?v=vOJfArcQ42A  

MUITO BOM: Jair Bolsonaro x Nando Moura. https://www.youtube.com/watch?v=MW_3JNHNl-4  

Antonia Fontenelle entrevista Jair Bolsonaro. https://www.youtube.com/watch?v=Mg-Q2wphBOM  

Bolsonaro em entrevista com a competente jornalista Leda Nagle. https://www.youtube.com/watch?v=1CpPj66Iy1o  

VAMOS ACREDITAR NO BRASIL.  ESSA É A NOSSA PÁTRIA. https://www.youtube.com/watch?v=xvkHF1qRgcw  

Dos EUA Jones Sathler fala sobre o Brasil e Trump: https://www.youtube.com/watch?v=0IrHoLiDpkE  

BAND: Bolsonaro é questionado por 5 jornalistas simultaneamente. https://www.youtube.com/watch?v=3bPgmvdz4yk  

O "Centrão" e algumas de suas pr

BOLSONARO NO BARRA SHOPPING https://www.youtube.com/watch?v=cg7t-N6M8q4 Agora à tarde no Barra Shopping  - Rio/RJ.

Grato a todos pelo carinho e consideração.

Brasil acima de tudo. Deus acima de todos! 

Globo reconhece, a Direita já tem seu representante: Jair Bolsonaro. https://www.youtube.com/watch?v=wwD3XCXHKaw  

O turismo no Brasil passa também por uma nova política ambiental. https://www.youtube.com/watch?v=bsSfu4o46us  

GRATO PELO CARINHO E CONSIDERAÇÃO https://www.youtube.com/watch?v=Ufgnr8320Dg Desembarque Deputado Jair Bolsonaro no Aeroporto do Galeão, Rio de Janeiro. 

MARIGHELLA, AINDA VIVO NA CABEÇA DE DESOCUPADOS. https://www.youtube.com/watch?v=uAsOnxvkm6A  

STF GARANTE COMBATE A CORRUPÇÃO https://www.youtube.com/watch?v=1MYMTbV7ZUU SUPREMO TRIBUNAL FEDERAL SUSPENDE A TRAMITAÇÃO DE EMENDA QUE CRIMINALIZA AÇÃO DE JUÍZES E MEMBROS DO MINISTÉRIO PÚBLICO, via mandado de segurança de autoria do Deputado Federal (SP) Eduardo Bolsonaro 

ALEXANDRE GARCIA E O ÚLTIMO PRESIDEN

In [62]:
Vids_Jair = []

for i in videosJ:
    ChannelTitle = i["snippet"]['channelTitle']
    PublishedAt = i["snippet"]['publishedAt']
    Title = i["snippet"]['title']
    Description = i["snippet"]['description'] 
    Id = i['snippet']['resourceId']['videoId']
    Link = "https://www.youtube.com/watch?v=" +i['snippet']['resourceId']['videoId']
    Vids_Jair.append((ChannelTitle, PublishedAt, Title, Description, Id, Link))
    
Vids_Jair = pd.DataFrame(Vids_Jair, columns = ['ChannelTitle', 'PublishedAt','Title', 'Description', 'Id', 'Link'])

Vids_Jair.PublishedAt = pd.to_datetime(Vids_Jair.PublishedAt) 

Vids_Jair.head(5)

Vids_Jair.to_csv('[PD]VD_JairB_30072020.csv')

In [61]:
Vids_Jair.loc[Vids_Jair.Id=="jg47rYougAs"]


ChannelTitle               PublishedAt  \
740  Jair Bolsonaro 2019-05-24 01:52:39+00:00   

                                                 Title  \
740  "Stand-up" de Jair Bolsonaro - A verdade com b...   

                                           Description           Id  \
740  Vídeo inusitado onde o Presidente Jair Bolsona...  jg47rYougAs   

                                            Link  
740  https://www.youtube.com/watch?v=jg47rYougAs

**Stats**

In [63]:
video_ids = list(map(lambda x:x ['snippet']['resourceId']['videoId'], videosJ))

def get_vid_stats (video_ids):
    stats = []
    for i in range(0, len(video_ids), 50):
        res = youtube.videos().list(id=",".join(video_ids[i:i+50]),
                                   part='statistics').execute()
        
        stats += res['items']
    
    return stats

In [64]:
stats = get_vid_stats(video_ids)
#stats_liked = sorted(stats, key=lambda x:int(x['statistics']['likeCount']), reverse=True)

In [65]:
stats_liked[0]

{'kind': 'youtube#video',
 'etag': 'PMd11IIit7bb9vgintqJXrvGCo8',
 'id': 'yVb1PHdqVMM',
 'statistics': {'viewCount': '2992130',
  'likeCount': '228383',
  'dislikeCount': '5799',
  'favoriteCount': '0',
  'commentCount': '11439'}}

In [68]:
for i in stats:
    print(i['statistics']['likeCount'], 
          i['statistics']['dislikeCount'],
          i['statistics']['commentCount'],
          i['id'],
          "https://www.youtube.com/watch?v="+i['id'])

6497 34 557 BPUyiUUpEcg https://www.youtube.com/watch?v=BPUyiUUpEcg
22814 107 3019 lIQMZRY_IDs https://www.youtube.com/watch?v=lIQMZRY_IDs
10022 64 752 4lGv7PCgbW4 https://www.youtube.com/watch?v=4lGv7PCgbW4
13120 118 1340 Z7c-gbhKmSk https://www.youtube.com/watch?v=Z7c-gbhKmSk
14128 58 1456 lt2IIg6p_MY https://www.youtube.com/watch?v=lt2IIg6p_MY
25679 96 2156 5f2wY39r7RI https://www.youtube.com/watch?v=5f2wY39r7RI
16551 63 1917 Ysnx4lTSshU https://www.youtube.com/watch?v=Ysnx4lTSshU
16923 58 1617 BJaKSw1fhJ0 https://www.youtube.com/watch?v=BJaKSw1fhJ0
52966 1231 3650 oVIJD_tuRPY https://www.youtube.com/watch?v=oVIJD_tuRPY
14900 59 993 XBp3cQEs4RE https://www.youtube.com/watch?v=XBp3cQEs4RE
8236 40 532 7t7SKjJFPjc https://www.youtube.com/watch?v=7t7SKjJFPjc
13511 54 1253 YMaA89YQ0IQ https://www.youtube.com/watch?v=YMaA89YQ0IQ
11060 59 844 fIa2ZRy_a_s https://www.youtube.com/watch?v=fIa2ZRy_a_s
14781 80 875 kI7HAFxe_k0 https://www.youtube.com/watch?v=kI7HAFxe_k0
3637 19 224 qYaABSqlzck 

21413 246 1741 q4V7OJje_No https://www.youtube.com/watch?v=q4V7OJje_No
7500 17 411 dPlFIg0mXBs https://www.youtube.com/watch?v=dPlFIg0mXBs
14907 74 1049 A-amWP6NTmw https://www.youtube.com/watch?v=A-amWP6NTmw
16567 106 2023 ClKhb44Bo-U https://www.youtube.com/watch?v=ClKhb44Bo-U
11016 51 836 EiLvg4nN524 https://www.youtube.com/watch?v=EiLvg4nN524
11163 69 555 1mNeUxKY5bY https://www.youtube.com/watch?v=1mNeUxKY5bY
13200 41 839 Y7tKx3zMTls https://www.youtube.com/watch?v=Y7tKx3zMTls
7887 126 535 96OARO3e3M0 https://www.youtube.com/watch?v=96OARO3e3M0
7802 35 490 i48IlCY7yzA https://www.youtube.com/watch?v=i48IlCY7yzA
28055 149 1737 CI-9Hn3Ii_g https://www.youtube.com/watch?v=CI-9Hn3Ii_g
30747 2024 5119 _3fFNbEbWpI https://www.youtube.com/watch?v=_3fFNbEbWpI
6250 22 450 IC7OgHjHZpw https://www.youtube.com/watch?v=IC7OgHjHZpw
108555 1925 8935 4exXeohgDtE https://www.youtube.com/watch?v=4exXeohgDtE
10980 215 771 Cyjv5ySLh3U https://www.youtube.com/watch?v=Cyjv5ySLh3U
19235 368 1444 eIoQdxW

11414 185 1114 pu2h7BQJz18 https://www.youtube.com/watch?v=pu2h7BQJz18
6844 86 713 QSMRCIaMoCc https://www.youtube.com/watch?v=QSMRCIaMoCc
9829 170 1056 mARMBWEwY2A https://www.youtube.com/watch?v=mARMBWEwY2A
8974 119 952 WUEgL7MCe9A https://www.youtube.com/watch?v=WUEgL7MCe9A
58855 379 4738 W6kE4REEY8U https://www.youtube.com/watch?v=W6kE4REEY8U
15778 150 706 1b8gPNMKY9k https://www.youtube.com/watch?v=1b8gPNMKY9k
6178 89 532 7A77t4iM-vQ https://www.youtube.com/watch?v=7A77t4iM-vQ
10755 52 604 hPsLcH8e-mU https://www.youtube.com/watch?v=hPsLcH8e-mU
30992 293 2060 zu7EC1jp-CE https://www.youtube.com/watch?v=zu7EC1jp-CE
12510 51 768 i1GX2nXST-c https://www.youtube.com/watch?v=i1GX2nXST-c
12459 214 1595 JsnvRd4zIa8 https://www.youtube.com/watch?v=JsnvRd4zIa8
7583 172 832 cbe3VFjyTEc https://www.youtube.com/watch?v=cbe3VFjyTEc
6672 72 536 w_edZoX4lCU https://www.youtube.com/watch?v=w_edZoX4lCU
16592 154 893 XqLFiZQ6lIY https://www.youtube.com/watch?v=XqLFiZQ6lIY
10922 83 740 mya8uFxAf0U h

KeyError: 'commentCount'

In [69]:
stats_jb = []

for i in stats:
    view = i['statistics']['viewCount']
    favorite = i['statistics']['favoriteCount']
    like = i['statistics']['likeCount'] 
    dislike = i['statistics']['dislikeCount']
    #commentCount = i['statistics']['commentCount']
    Id = i['id']
    link = "https://www.youtube.com/watch?v="+i['id']
    stats_jb.append((view, favorite, like, dislike, Id, link))
    

stats_jb = pd.DataFrame(stats_eb, columns = ['View','Favorite', 'Like', 'Dislike', "Id", 'Link'])


In [70]:
stats_jb.head(6)

View Favorite   Like Dislike           Id  \
0   38373        0  13097      66  2yjWkL9WGVU   
1   60435        0  16355     307  4-0SaTclpIk   
2   14213        0   4635      15  ntwubMpc_KA   
3   13180        0   5393      17  jRYWw76wobs   
4  114077        0  32627     354  VYFY4FASyRg   
5   15514        0   5478      13  VMqubSpjkYE   

                                          Link  
0  https://www.youtube.com/watch?v=2yjWkL9WGVU  
1  https://www.youtube.com/watch?v=4-0SaTclpIk  
2  https://www.youtube.com/watch?v=ntwubMpc_KA  
3  https://www.youtube.com/watch?v=jRYWw76wobs  
4  https://www.youtube.com/watch?v=VYFY4FASyRg  
5  https://www.youtube.com/watch?v=VMqubSpjkYE

In [71]:
stats_jb.to_csv('[PD]Jair_Stats2_3007202.csv.csv')

**Parenteses comentários de um vídeo**

In [250]:
req_playlist = youtube.playlistItems().list(videoId = "ybote10acL4", 
                                            part = "snippet", 
                                            maxResults = 50)

In [252]:
testecoment = youtube.commentThreads().list(videoId ="ybote10acL4", part = "snippet", maxResults = 50)
comment_J = testecoment.execute()

In [254]:
comment_J['items'][0]

{'kind': 'youtube#commentThread',
 'etag': 'hOyqDmtEocTqsHsPR4vFXX4Mez8',
 'id': 'Ugz0ObjxfqAkg-e7NIV4AaABAg',
 'snippet': {'videoId': 'ybote10acL4',
  'topLevelComment': {'kind': 'youtube#comment',
   'etag': '0qr4vh2gkG7pyqABbmRVix_46fc',
   'id': 'Ugz0ObjxfqAkg-e7NIV4AaABAg',
   'snippet': {'videoId': 'ybote10acL4',
    'textDisplay': 'Kkkkkkkk ta com corona besta cadê o seu histórico de atleta agora em? Kkkkkkkkkkkk',
    'textOriginal': 'Kkkkkkkk ta com corona besta cadê o seu histórico de atleta agora em? Kkkkkkkkkkkk',
    'authorDisplayName': 'contra os Corintianos ladrão',
    'authorProfileImageUrl': 'https://yt3.ggpht.com/a/AATXAJzx9Sykd0D-vkj4QdbW_h6IH9kVPmgn9UvxDipi=s48-c-k-c0xffffffff-no-rj-mo',
    'authorChannelUrl': 'http://www.youtube.com/channel/UCiIyL89tBt2aELyaI4S0rEw',
    'authorChannelId': {'value': 'UCiIyL89tBt2aELyaI4S0rEw'},
    'canRate': True,
    'viewerRating': 'none',
    'likeCount': 0,
    'publishedAt': '2020-07-23T22:12:13Z',
    'updatedAt': '2020-0

<h1>Flávio</h1>

In [72]:
q_FlvB = youtube.channels().list(id = "UCl2HptoHv6PjZMQAwTdA--Q", part = "contentDetails")
Chan_FlvB = q_FlvB.execute()

In [257]:
Chan_FlvB

{'kind': 'youtube#channelListResponse',
 'etag': 'IWWvR4Vo7e3WiJ8yxWVJfAbEzr8',
 'pageInfo': {'totalResults': 1, 'resultsPerPage': 1},
 'items': [{'kind': 'youtube#channel',
   'etag': 'hDKqsjX2j4uo2dJ1TGLUOkRTh-8',
   'id': 'UCl2HptoHv6PjZMQAwTdA--Q',
   'contentDetails': {'relatedPlaylists': {'likes': '',
     'favorites': '',
     'uploads': 'UUl2HptoHv6PjZMQAwTdA--Q',
     'watchHistory': 'HL',
     'watchLater': 'WL'}}}]}

***Content Details***

- items
 - 'id': 'UCl2HptoHv6PjZMQAwTdA--Q'
 - 'uploads': 'UUl2HptoHv6PjZMQAwTdA--Q'

**50 vids**

In [281]:
q_flv_chn = youtube.playlistItems().list(playlistId = "UUl2HptoHv6PjZMQAwTdA--Q", part = "snippet", maxResults = 50).execute()

In [282]:
q_flv_chn['items'][0]

{'kind': 'youtube#playlistItem',
 'etag': 'Lr_nuMJzSA8XdoCeW-sZYdGwBbc',
 'id': 'VVVsMkhwdG9IdjZQalpNUUF3VGRBLS1RLkZud2NuVjZRT2Zr',
 'snippet': {'publishedAt': '2020-07-24T03:24:46Z',
  'channelId': 'UCl2HptoHv6PjZMQAwTdA--Q',
  'title': 'Live semanal do presidente Jair Bolsonaro - 23.07.20',
  'description': '',
  'thumbnails': {'default': {'url': 'https://i.ytimg.com/vi/FnwcnV6QOfk/default.jpg',
    'width': 120,
    'height': 90},
   'medium': {'url': 'https://i.ytimg.com/vi/FnwcnV6QOfk/mqdefault.jpg',
    'width': 320,
    'height': 180},
   'high': {'url': 'https://i.ytimg.com/vi/FnwcnV6QOfk/hqdefault.jpg',
    'width': 480,
    'height': 360},
   'standard': {'url': 'https://i.ytimg.com/vi/FnwcnV6QOfk/sddefault.jpg',
    'width': 640,
    'height': 480},
   'maxres': {'url': 'https://i.ytimg.com/vi/FnwcnV6QOfk/maxresdefault.jpg',
    'width': 1280,
    'height': 720}},
  'channelTitle': 'Flavio Bolsonaro',
  'playlistId': 'UUl2HptoHv6PjZMQAwTdA--Q',
  'position': 0,
  'resourceId

In [283]:
#explorar
for r in q_flv_chn['items']:
    print(r['snippet']['title'], r['snippet']['description'],"https://www.youtube.com/watch?v="+r['snippet']['resourceId']['videoId'])

Live semanal do presidente Jair Bolsonaro - 23.07.20  https://www.youtube.com/watch?v=FnwcnV6QOfk
Globo não está preocupada com vidas, mas sim com $$$ e derrubar Bolsonaro. Jornalista Rica Perrone desnuda a hipocrisia da Globo, materializada no aspirador (a comentarista, claro!) Walter Casagrande.
Quando a Globo livra com o jogo ela fala bem (campeonato paulista), mas quando o jogo é transmitido pelo SBT ela faz matéria fúnebre (campeonato carioca, final Fla x Flu). https://www.youtube.com/watch?v=MuyYNbm-aE4
Saiba a verdade! Esqueça os especialistas midiáticos. A grande mídia (você sabe quem), não tá preocupado com sua saúde! Ela é entusiasta do pânico e do caos!!

Saiba a verdade... Assista!!! https://www.youtube.com/watch?v=z-KtjgULdtM
Entrega de sistemas adutores e equipamentos de dessalinização em Alagoas Também liberamos recursos para obras de saneamento e abastecimento de água. As ações do governo do presidente Jair Bolsonaro vão melhorar a qualidade de vida e gerar mai

In [288]:
#salvar
Titulos_Flv = []

for r in q_flv_chn['items']:
    Title = r['snippet']['title']
    Description = r['snippet']['description']
    Link = "https://www.youtube.com/watch?v="+r['snippet']['resourceId']['videoId']
    Titulos_Flv.append((Title, Description,  Link))
    
Titulos_Flv = pd.DataFrame(Titulos_Flv, columns = ['Title', 'Description',  'Link'])


Titulos_Flv.to_csv('[PD]VD_Flavio_50TitComent.csv')

#Titulos_Flv.head(10)

**Todos os Vids**

In [73]:
#'id': 'UCl2HptoHv6PjZMQAwTdA--Q'

videos = get_channel_videos('UCl2HptoHv6PjZMQAwTdA--Q')
len(videos)

919

In [74]:
videos[0]

{'kind': 'youtube#playlistItem',
 'etag': '3U7dzEgwr2__37TQM5x1sECCoyA',
 'id': 'VVVsMkhwdG9IdjZQalpNUUF3VGRBLS1RLjhtZ1pZTVdGc2xJ',
 'snippet': {'publishedAt': '2020-07-30T14:14:19Z',
  'channelId': 'UCl2HptoHv6PjZMQAwTdA--Q',
  'title': 'Presidente Bolsonaro sendo recebido no aeroporto de São Raimundo Nonato - PIAUÍ - 30.07.20',
  'description': '',
  'thumbnails': {'default': {'url': 'https://i.ytimg.com/vi/8mgZYMWFslI/default.jpg',
    'width': 120,
    'height': 90},
   'medium': {'url': 'https://i.ytimg.com/vi/8mgZYMWFslI/mqdefault.jpg',
    'width': 320,
    'height': 180},
   'high': {'url': 'https://i.ytimg.com/vi/8mgZYMWFslI/hqdefault.jpg',
    'width': 480,
    'height': 360}},
  'channelTitle': 'Flavio Bolsonaro',
  'playlistId': 'UUl2HptoHv6PjZMQAwTdA--Q',
  'position': 0,
  'resourceId': {'kind': 'youtube#video', 'videoId': '8mgZYMWFslI'}}}

In [302]:
# analizar: título, link, descrição \n 

for r in videos:
    print(r['snippet']['title'], "https://www.youtube.com/watch?v="+r['snippet']['resourceId']['videoId'], r['snippet']['description'], "\n")

Live semanal do presidente Jair Bolsonaro - 23.07.20 https://www.youtube.com/watch?v=FnwcnV6QOfk  

Globo não está preocupada com vidas, mas sim com $$$ e derrubar Bolsonaro. https://www.youtube.com/watch?v=MuyYNbm-aE4 Jornalista Rica Perrone desnuda a hipocrisia da Globo, materializada no aspirador (a comentarista, claro!) Walter Casagrande.
Quando a Globo livra com o jogo ela fala bem (campeonato paulista), mas quando o jogo é transmitido pelo SBT ela faz matéria fúnebre (campeonato carioca, final Fla x Flu). 

Saiba a verdade! Esqueça os especialistas midiáticos. https://www.youtube.com/watch?v=z-KtjgULdtM A grande mídia (você sabe quem), não tá preocupado com sua saúde! Ela é entusiasta do pânico e do caos!!

Saiba a verdade... Assista!!! 

Entrega de sistemas adutores e equipamentos de dessalinização em Alagoas https://www.youtube.com/watch?v=uU9FErd8Z1s Também liberamos recursos para obras de saneamento e abastecimento de água. As ações do governo do presidente Jair Bols


AULA DE CIDADANIA E PATRIOTISMO EM LIBRAS https://www.youtube.com/watch?v=Hrx8XFRP_tM #EstouComBolsonaro #Bolsonaro17 #Libras
------------------------------------------------------------------------------------------------
INSCREVA-SE NO CANAL: https://www.youtube.com/user/canalBOLSONARO

REDES SOCIAIS:

- FACEBOOK: https://www.facebook.com/flaviobolsonaro
- TWITTER: https://twitter.com/flaviobolsonaro
- INSTAGRAM: https://www.instagram.com/flaviobolsonaro
- SNAPCHAT: flaviobolsonaro

- SITE: www.flaviobolsonaro.com 

DE FAKE NEWS E CORRUPÇÃO O PT ENTENDE BEM! https://www.youtube.com/watch?v=FyUfPNuo2mI ------------------------------------------------------------------------------------------------
INSCREVA-SE NO CANAL: https://www.youtube.com/user/canalBOLSONARO

REDES SOCIAIS:

- FACEBOOK: https://www.facebook.com/flaviobolsonaro
- TWITTER: https://twitter.com/flaviobolsonaro
- INSTAGRAM: https://www.instagram.com/flaviobolsonaro
- SNAPCHAT: flaviobolsonaro

- SITE: www.flaviobolson


Bombeiros e Policiais: A LUTA CONTINUA! https://www.youtube.com/watch?v=zrHb4cdJKcI  

BOLSONARO - Militar é preso e depois tem que se virar para provar inocência. https://www.youtube.com/watch?v=NZBsOd0ccR4  

Bolsonaro - Nem Fundo de Combate a Pobreza e nem Cotas! Chega de enganação!!! https://www.youtube.com/watch?v=KDy9U1Scpmo  

Bolsonaro lança Frente Parlamentar pela Transparência na Gestão Pública https://www.youtube.com/watch?v=iBFHuDYc2FA  

Flávio Bolsonaro evidencia falhas na condução de concursos públicos pela SEAP https://www.youtube.com/watch?v=pHSQYrp9NwU  

Com discurso de Bolsonaro PEC 23 não é aprovada. Deputados favoráveis abandonam plenário https://www.youtube.com/watch?v=Lh0zhn-_HNU  

BOLSONARO REBATE ARGUMENTOS DE GOVERNISTAS E SEGUE NA DEFESA DOS BOMBEIROS. https://www.youtube.com/watch?v=a54kr9gr5lc  

BOLSONARO desmascara proposta inicial do Governo https://www.youtube.com/watch?v=s1Vbwgxsk1o  

Bolsonaro contra a industria do "dano moral gay" https://www.you

In [90]:
Vids_Flv = []

for i in videos:
    ChannelTitle = i["snippet"]['channelTitle']
    PublishedAt = i["snippet"]['publishedAt']
    Title = i["snippet"]['title']
    Description = i["snippet"]['description'] 
    Id = i['snippet']['resourceId']['videoId']
    Link = "https://www.youtube.com/watch?v=" +i['snippet']['resourceId']['videoId']
    Vids_Flv.append((ChannelTitle, PublishedAt, Title, Description, Id, Link))
    
Vids_Flv = pd.DataFrame(Vids_Flv, columns = ['ChannelTitle', 'PublishedAt','Title', 'Description', 'Id', 'Link'])

Vids_Flv.PublishedAt = pd.to_datetime(Vids_Flv.PublishedAt) 

Vids_Flv.head(5)

Vids_Flv.to_csv('[PD]VD_FlvB_30072020.csv')

In [89]:
Vids_Flv.loc[Vids_Flv.Id=="AkKKJfuFQBQ"]

ChannelTitle               PublishedAt  \
346  Flavio Bolsonaro 2019-05-28 14:11:51+00:00   

                                                 Title  \
346  "STAND-UP" DE JAIR BOLSONARO - A VERDADE COM B...   

                                           Description           Id  \
346  #PresidenteJairBolsonaro\n\nVia: Canal Folha d...  AkKKJfuFQBQ   

                                            Link  
346  https://www.youtube.com/watch?v=AkKKJfuFQBQ

**Stats**

In [91]:
video_ids = list(map(lambda x:x ['snippet']['resourceId']['videoId'], videos))

def get_vid_stats (video_ids):
    stats = []
    for i in range(0, len(video_ids), 50):
        res = youtube.videos().list(id=",".join(video_ids[i:i+50]),
                                   part='statistics').execute()
        
        stats += res['items']
    
    return stats

In [92]:
stats = get_vid_stats(video_ids)
#stats_liked = sorted(stats, key=lambda x:int(x['statistics']['likeCount']), reverse=True)

In [97]:
stats[0]

{'kind': 'youtube#video',
 'etag': '8OmN-D-LIgR-mbZXFZv11DKK1vw',
 'id': '8mgZYMWFslI',
 'statistics': {'viewCount': '14167',
  'likeCount': '4428',
  'dislikeCount': '27',
  'favoriteCount': '0',
  'commentCount': '968'}}

In [98]:
for i in stats :
    print(i['statistics']['likeCount'], 
          i['statistics']['dislikeCount'],
          i['statistics']['commentCount'],
          i['id'],
          "https://www.youtube.com/watch?v="+i['id'])

4428 27 968 8mgZYMWFslI https://www.youtube.com/watch?v=8mgZYMWFslI
1464 4 142 JbMyAFmJE78 https://www.youtube.com/watch?v=JbMyAFmJE78
2041 12 185 SJhPF9qlcdE https://www.youtube.com/watch?v=SJhPF9qlcdE
1320 4 120 -SE3u4smQ5Y https://www.youtube.com/watch?v=-SE3u4smQ5Y
1859 6 198 CW5Ca5QVIEo https://www.youtube.com/watch?v=CW5Ca5QVIEo
2097 8 162 VXLzU0wxPjg https://www.youtube.com/watch?v=VXLzU0wxPjg
1802 6 187 jUK0iLO0CLw https://www.youtube.com/watch?v=jUK0iLO0CLw
1114 0 133 izbIg7cCMas https://www.youtube.com/watch?v=izbIg7cCMas
1700 14 231 Z7Y25Y8bvOk https://www.youtube.com/watch?v=Z7Y25Y8bvOk
1112 7 182 FnwcnV6QOfk https://www.youtube.com/watch?v=FnwcnV6QOfk
8402 116 895 MuyYNbm-aE4 https://www.youtube.com/watch?v=MuyYNbm-aE4
475 1 34 z-KtjgULdtM https://www.youtube.com/watch?v=z-KtjgULdtM
1388 3 151 uU9FErd8Z1s https://www.youtube.com/watch?v=uU9FErd8Z1s
1822 7 260 UAtgxXXsepE https://www.youtube.com/watch?v=UAtgxXXsepE
821 3 69 QlraUYesMQE https://www.youtube.com/watch?v=QlraUY

1271 11 128 ut4gkzxGZnY https://www.youtube.com/watch?v=ut4gkzxGZnY
1076 5 82 MlRZzSQ8_TE https://www.youtube.com/watch?v=MlRZzSQ8_TE
995 5 58 GiYHu46DkCw https://www.youtube.com/watch?v=GiYHu46DkCw
1193 16 101 xON11No603Q https://www.youtube.com/watch?v=xON11No603Q
1183 7 80 qTLveT7xs9o https://www.youtube.com/watch?v=qTLveT7xs9o
1405 12 124 9TUkJpgF3jg https://www.youtube.com/watch?v=9TUkJpgF3jg
997 24 101 7z6GZ6tmnG0 https://www.youtube.com/watch?v=7z6GZ6tmnG0
1444 20 154 UA0a_Dfp4sY https://www.youtube.com/watch?v=UA0a_Dfp4sY
1211 25 152 dv1E_SowGks https://www.youtube.com/watch?v=dv1E_SowGks
1020 14 144 _txSnsoG23U https://www.youtube.com/watch?v=_txSnsoG23U
632 10 63 FyFGtqHjFwg https://www.youtube.com/watch?v=FyFGtqHjFwg
624 8 82 VMSBgAY2j28 https://www.youtube.com/watch?v=VMSBgAY2j28
1463 9 188 fpgOnAUWngQ https://www.youtube.com/watch?v=fpgOnAUWngQ
2814 145 782 ICliGxdRB2g https://www.youtube.com/watch?v=ICliGxdRB2g
1821 23 193 N9--10A8kgk https://www.youtube.com/watch?v=N9--1

KeyError: 'commentCount'

In [99]:
# Salvar

stats_FLV = []

for i in stats:
    #commentCount = i['statistics']['commentCount']
    view = i['statistics']['viewCount']
    favorite = i['statistics']['favoriteCount']
    like = i['statistics']['likeCount'] 
    dislike = i['statistics']['dislikeCount']
    i['id']
    link = ("https://www.youtube.com/watch?v="+i['id'])
    stats_FLV.append((view, favorite, like, dislike, Id, link))
    

stats_FLV = pd.DataFrame(stats_FLV, columns = ['View','Favorite', 'Like', 'Dislike', "Id", 'Link'])



stats_FLV.to_csv('[PD]Flavio_Stats2_3007202.csv')


<h1> Comentários de um vídeo </h1>


"Stand-up" de Jair Bolsonaro - A verdade com bom humor https://www.youtube.com/watch?v=jg47rYougAs Vídeo inusitado onde o Presidente Jair Bolsonaro fala, com muito bom humor, sobre  a realidade dos fatos e as Fake News. 

id = jg47rYougAs

In [3]:
q_vid_comment = youtube.commentThreads().list(videoId ="jg47rYougAs", part = "snippet", maxResults = 50)

comment_J001 = q_vid_comment.execute()

In [40]:
comment_J001['items'][49]

{'kind': 'youtube#commentThread',
 'etag': 'wjuvghpDzL-nt4t41uDRNIk8rKM',
 'id': 'UgyiWAapW2YA26q41aN4AaABAg',
 'snippet': {'videoId': 'jg47rYougAs',
  'topLevelComment': {'kind': 'youtube#comment',
   'etag': 'BSDGHeOrU_0I-9kl8VYEE9YngEY',
   'id': 'UgyiWAapW2YA26q41aN4AaABAg',
   'snippet': {'videoId': 'jg47rYougAs',
    'textDisplay': '🤣🤣🤣🤣🤣',
    'textOriginal': '🤣🤣🤣🤣🤣',
    'authorDisplayName': 'MARI DRUMOND',
    'authorProfileImageUrl': 'https://yt3.ggpht.com/a/AATXAJxITOFwySuyIPzzQDNz_MyoKRGMOkGqSNQLmA=s48-c-k-c0xffffffff-no-rj-mo',
    'authorChannelUrl': 'http://www.youtube.com/channel/UCDOxchbAYTTfbxPQfxEl-iA',
    'authorChannelId': {'value': 'UCDOxchbAYTTfbxPQfxEl-iA'},
    'canRate': True,
    'viewerRating': 'none',
    'likeCount': 0,
    'publishedAt': '2019-05-28T15:43:38Z',
    'updatedAt': '2019-05-28T15:43:38Z'}},
  'canReply': True,
  'totalReplyCount': 0,
  'isPublic': True}}

In [35]:
for r in comment_J001['items']:
    print("author: " +r['snippet']['topLevelComment']['snippet']['authorDisplayName'] + " ",
          " - " +r['snippet']['topLevelComment']['snippet']['textDisplay'], 
          "\n")

author: Cripto Is the Future   - Canalhas, canalhas ,hahaha, takei. 

author: Alessandra Abreu Rodrigues Vieira   - <a href="http://www.youtube.com/results?search_query=%23bolsonaro2022">#Bolsonaro2022</a> 

author: Ana Paula Rodrigues Conceição   - 🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮🤮 

author: Ana Paula Rodrigues Conceição   - Pessimo 🤮🤬🤬🤬🤬🤬 

author: 11 de Setembro 2021   - Vc ainda sonha em dar pro Trump? 

author: Yogan Kupas   - Kkk 

author: davison filme   - Fala do dinheiro pago ao sbt a recor e band 

author: davison filme   - Fala da interferência na polícia federal,  pra defender o Queiroz e seus filhos. 

author: davison filme   - Fala dos funcionários laranjas,  onde estão? 

author: davison filme   - Fala das rachadinhas,  como que funciona? 

author: davison filme   - Kd os cheques da Michele? 

author: davison filme   - Mostra seu telefone bozo 

author: jonatas neto   - Mitoo 🤣🤣👏👏👏👏👏 

author: Felipe Dias   - ASSASSINO do povo seu lixo 

author: Flavia Lemonie   - Babacao 



In [1]:
q_vid_comment3 = youtube.commentThreads().list(videoId ="jg47rYougAs", part = "snippet",  order = "relevance", maxResults = 100)

comment_J003 = q_vid_comment3.execute()

NameError: name 'youtube' is not defined

In [79]:
comment_J003['items'][5]

{'kind': 'youtube#commentThread',
 'etag': 'X1PK6-cAwuYSJnDosQ3lJPZDJx0',
 'id': 'Ugyti96BYoVN2pWYcHd4AaABAg',
 'snippet': {'videoId': 'jg47rYougAs',
  'topLevelComment': {'kind': 'youtube#comment',
   'etag': 'R8b5woMZYnTNQTDjOLOFYP2iNCE',
   'id': 'Ugyti96BYoVN2pWYcHd4AaABAg',
   'snippet': {'videoId': 'jg47rYougAs',
    'textDisplay': 'Meu sonho virou realidade ? Stand up do Bolsonaro?',
    'textOriginal': 'Meu sonho virou realidade ? Stand up do Bolsonaro?',
    'authorDisplayName': 'Richteru',
    'authorProfileImageUrl': 'https://yt3.ggpht.com/a/AATXAJzL2sSUh0j3yLroPO32uhBWMuphxqDvm66SuPiZ=s48-c-k-c0xffffffff-no-rj-mo',
    'authorChannelUrl': 'http://www.youtube.com/channel/UCYKolLcvVjavQN64-gRP9Sg',
    'authorChannelId': {'value': 'UCYKolLcvVjavQN64-gRP9Sg'},
    'canRate': True,
    'viewerRating': 'none',
    'likeCount': 828,
    'publishedAt': '2019-05-24T01:53:34Z',
    'updatedAt': '2019-05-24T01:53:34Z'}},
  'canReply': True,
  'totalReplyCount': 12,
  'isPublic': True

In [95]:
#Explorar

for r in comment_J003['items']:
    print(r['snippet']['topLevelComment']['snippet']['publishedAt'],"-",
          r['snippet']['topLevelComment']['snippet']['authorDisplayName'],
          r['snippet']['topLevelComment']['snippet']['authorChannelUrl'],
          r['snippet']['topLevelComment']['snippet']['likeCount'], "-L/",
          r['snippet']['totalReplyCount'],  "-R/" , "\n",
          r['snippet']['topLevelComment']['snippet']['textOriginal'], 
          "\n")

2019-05-24T01:58:56Z - Myrian Mattos http://www.youtube.com/channel/UCJd8G9EIfUZL5GV-y-_hLuw 76 -L/ 0 -R/ 
 Amamos o senhor Presidente Bolsonaro e  dia 26 VAMOS abarrotar às ruas defendendo o Brasil e o seu HONESTO governo! Amém! 

2019-05-24T06:42:19Z - Joao Vitor http://www.youtube.com/channel/UCJoRyy9av5tnUZguUasNt3g 146 -L/ 0 -R/ 
 Kkkkkkkk melhor stand -up que eu já vi, esse Presidente é demais. 

2019-05-24T04:00:17Z - Dylan Farias http://www.youtube.com/channel/UCS6DsmPhJsIxdLrFiMx5MBA 195 -L/ 2 -R/ 
 *_Mesmo Com Toda _**_#Conspiração_**_ De Traidores Contra Esse "_**_#Herói_**_", Ele Ainda Tira Tempo Para _**_#Sorrir_**_ E Melhor Ainda, Nos Fazer _**_#Sorrir_**_... _**_#TeAmo_**_ Cara!_* 

2019-05-24T03:56:26Z - Dylan Farias http://www.youtube.com/channel/UCS6DsmPhJsIxdLrFiMx5MBA 79 -L/ 3 -R/ 
 *_Tiririca, Whinderson,Tom Cavalcante, Gregório Duvivier, Rafinha Bastos, Fábio Porchat, _**_#Aprendam_**_ À Fazerem "_**_#StandUp_**_" Com O Mito!_* 

2019-05-24T11:35:49Z - Maria Apare

In [99]:
#Salvar

Coment_jg47rYougAs = []

for r in comment_J003['items']:
    PublishedAt = r['snippet']['topLevelComment']['snippet']['publishedAt']
    Name = r['snippet']['topLevelComment']['snippet']['authorDisplayName']
    AuthorChannelUrl = r['snippet']['topLevelComment']['snippet']['authorChannelUrl']
    Like = r['snippet']['topLevelComment']['snippet']['likeCount']
    Reply = r['snippet']['totalReplyCount']
    Text = r['snippet']['topLevelComment']['snippet']['textOriginal']
    Coment_jg47rYougAs.append((PublishedAt, Name, AuthorChannelUrl, Like, Reply, Text))
    

In [101]:
Coment_jg47rYougAs= pd.DataFrame(Coment_jg47rYougAs, columns= ["PublishedAt", "Name", "AuthorChannelUrl", "Like",  "Reply", "Text"])
Coment_jg47rYougAs.head(10)

PublishedAt                     Name  \
0  2019-05-24T01:58:56Z            Myrian Mattos   
1  2019-05-24T06:42:19Z               Joao Vitor   
2  2019-05-24T04:00:17Z             Dylan Farias   
3  2019-05-24T03:56:26Z             Dylan Farias   
4  2019-05-24T11:35:49Z          Maria Aparecida   
5  2019-05-24T01:53:34Z                 Richteru   
6  2019-05-24T05:02:00Z           Isaias Firmino   
7  2019-05-24T09:24:29Z          Regina Mazzocco   
8  2019-05-24T06:11:11Z  Verônica da Costa Lemos   
9  2019-05-24T03:40:05Z                  Ali' en   

                                    AuthorChannelUrl  Like  Reply  \
0  http://www.youtube.com/channel/UCJd8G9EIfUZL5G...    76      0   
1  http://www.youtube.com/channel/UCJoRyy9av5tnUZ...   146      0   
2  http://www.youtube.com/channel/UCS6DsmPhJsIxdL...   195      2   
3  http://www.youtube.com/channel/UCS6DsmPhJsIxdL...    79      3   
4  http://www.youtube.com/channel/UCPWM-k2C7FTGQ3...   109      0   
5  http://www.youtube.com/channel/UCYKolLcvVjavQN...   828     12   
6  http://www.youtube.com/channel/UC4L7g9Pki-w6j4...    50      1   
7  http://www.youtube.com/channel/UCnsc5RPUGRgMgG...    20      1   
8  http://www.youtube.com/channel/UC3LPETw857QHp4...    24      0   
9  http://www.youtube.com/channel/UCdVjBKlCmAaGp6...    19      1   

                                                Text  
0  Amamos o senhor Presidente Bolsonaro e  dia 26...  
1  Kkkkkkkk melhor stand -up que eu já vi, esse P...  
2  *_Mesmo Com Toda _**_#Conspiração_**_ De Traid...  
3  *_Tiririca, Whinderson,Tom Cavalcante, Gregóri...  
4  DOMINGO, DIA 26 TODOS GRITANDO PACIFICAMENTE, ...  
5  Meu sonho virou realidade ? Stand up do Bolson...  
6  Votei Bolsonaro 2018 votarei 2022 novamente.. B17  
7  **ORAR PELO  BRASIL**\n\n“ASSUMA O COMPROMISSO...  
8  O que é um Mito?? ...\nBolsonaro o Nosso Presi...  
9  Finalmente um Presidente que me faz sorrir e n...

In [102]:
Coment_jg47rYougAs["PublishedAt"] = pd.to_datetime(Coment_jg47rYougAs["PublishedAt"])
#Coment_jg47rYougAs.to_csv("Coment_Stand-up.csv")

In [139]:
Coment_jg47rYougAs.tail(15)

PublishedAt  \
85 2019-05-24 01:54:47+00:00   
86 2019-05-24 04:09:04+00:00   
87 2019-05-24 11:15:04+00:00   
88 2019-05-24 04:12:37+00:00   
89 2019-05-24 05:10:58+00:00   
90 2019-05-24 01:53:57+00:00   
91 2019-05-24 09:37:42+00:00   
92 2019-05-25 00:23:04+00:00   
93 2019-05-24 14:15:16+00:00   
94 2019-05-24 10:11:40+00:00   
95 2019-05-24 02:06:35+00:00   
96 2019-05-24 12:01:37+00:00   
97 2019-05-24 15:56:32+00:00   
98 2019-05-25 01:44:44+00:00   
99 2019-05-24 10:02:32+00:00   

                                                 Name  \
85                                    Marcus Vinícius   
86                                             Mateus   
87                                       Drica Brasil   
88                                    Cristiano Gilvn   
89                                     Sileimar Lopes   
90                                     Mateus Romeiro   
91                                   dicas de costura   
92  Treinamento P habilitados Z Leste Suzano Itaqu...   
93  Sempre que você ler os comentários eu vou está lá   
94                                     noan moonraker   
95                       Darlan Frederico de Oliveira   
96                            Dercio Moreno Mancoelho   
97                                      Suellen Silva   
98                                   Família Carvalho   
99                                         Neuza Reis   

                                     AuthorChannelUrl  Like  Reply  \
85  http://www.youtube.com/channel/UC01Y2iS0xR8vcT...   272     21   
86  http://www.youtube.com/channel/UCYxWptlLc7KUlV...     0      0   
87  http://www.youtube.com/channel/UCj2b4jDCNgEv8J...     1      0   
88  http://www.youtube.com/channel/UCQT5HDaYmi1lYT...     0      0   
89  http://www.youtube.com/channel/UC-Vb1VfIa9Kj1E...     4      0   
90  http://www.youtube.com/channel/UCs0tyjCxmKXzPO...   195      0   
91  http://www.youtube.com/channel/UCzt3v5gaNRQnzo...     0      0   
92  http://www.youtube.com/channel/UCn_sgGTahl0p5H...     3      0   
93  http://www.youtube.com/channel/UCrWYwYQEM4Qs0a...     2      0   
94  http://www.youtube.com/channel/UCj1ege9153eSqn...     0      0   
95  http://www.youtube.com/channel/UC6hNscFMaB_9z0...   110      0   
96  http://www.youtube.com/channel/UC6usaFrguVbk4U...     1      0   
97  http://www.youtube.com/channel/UCs1q4gzXDcnWAj...     2      0   
98  http://www.youtube.com/channel/UCKMsmbmEYlqgvP...     3      0   
99  http://www.youtube.com/channel/UCdUTBSom_Tbnd5...     0      0   

                                                 Text  
85  Bolsonaro,  estarei na rua dia 26 pro seu Gove...  
86  DIA 26/05/2019 MANIFESTAÇOES EM TODO BRASIL, E...  
87  Muito orgulho do nosso presidente 👏👏👏 Ele não ...  
88  Quem também quer ver mais vídeos assim no cana...  
89  Parabéns presidente, continue firme e conta co...  
90  O melhor presidente! Forte Abraço hahahahahaha...  
91                                 Quem vai dia 26 ❓❓  
92  Sorriso e 5 minutos no horário gratuito não me...  
93                        Sou Bolsominion de coração!  
94           Como eu amo o senhor meu presidente!!❤🇧🇷  
95  Ficou top o Stand-up do Jair Bolsonaro, ri dem...  
96  É tanto Blá Blá que ouvi nesse vídeo do link a...  
97  Sério que tem gente que não ama esse cara!!! T...  
98  Parabéns meu Presidente! Sinto muito orgulho d...  
99                   Além de honesto é divertido\n😂😂😂

In [145]:
#Coment_jg47rYougAs.shape
#Coment_jg47rYougAs.dtypes

#Coment_jg47rYougAs.PublishedAt.unique()
#Coment_jg47rYougAs.Name.nunique()


#Coment_jg47rYougAs.Like.describe()
#Coment_jg47rYougAs.Name.value_counts()

#%matplotlib inline
#Coment_jg47rYougAs.Like.plot(kind="hist")

#Coment_jg47rYougAs.Like.sort_values(ascending=False)

Coment_jg47rYougAs.sort_values("Reply", ascending=False)
#Coment_jg47rYougAs.sort_values("Like", ascending=False)
#Coment_jg47rYougAs.sort_values(["Like", "Reply"], ascending=False)


PublishedAt                     Name  \
85 2019-05-24 01:54:47+00:00          Marcus Vinícius   
5  2019-05-24 01:53:34+00:00                 Richteru   
10 2019-05-24 01:54:51+00:00             Matheus Reis   
15 2019-05-24 01:54:10+00:00               Gabriel C.   
32 2019-05-24 04:06:56+00:00         Koetsu Takahashi   
..                       ...                      ...   
39 2019-05-24 07:15:47+00:00  Robson Da Silva Bezerra   
38 2019-05-24 10:23:24+00:00    Maria jose F. Teofilo   
37 2019-05-24 13:18:32+00:00             Wayne Colaço   
35 2019-05-24 02:06:28+00:00    Marcos Andre Oliveira   
99 2019-05-24 10:02:32+00:00               Neuza Reis   

                                     AuthorChannelUrl  Like  Reply  \
85  http://www.youtube.com/channel/UC01Y2iS0xR8vcT...   272     21   
5   http://www.youtube.com/channel/UCYKolLcvVjavQN...   828     12   
10  http://www.youtube.com/channel/UCgcsyKjt0r8P2F...   702      7   
15  http://www.youtube.com/channel/UC0zCQJAifegYnI...    53      5   
32  http://www.youtube.com/channel/UC1-YZuSmAg_QO4...    45      3   
..                                                ...   ...    ...   
39  http://www.youtube.com/channel/UCekGdZiiVMYVNB...     5      0   
38  http://www.youtube.com/channel/UCNczDm5gUX83B3...    10      0   
37  http://www.youtube.com/channel/UCWKbH_2DQlQne8...    10      0   
35  http://www.youtube.com/channel/UCAC_Q8NjBXTBUw...    27      0   
99  http://www.youtube.com/channel/UCdUTBSom_Tbnd5...     0      0   

                                                 Text  
85  Bolsonaro,  estarei na rua dia 26 pro seu Gove...  
5   Meu sonho virou realidade ? Stand up do Bolson...  
10  Reparem que ele tá começando a discursar parec...  
15  Mesmo sendo presidente, Bolsonaro é melhor faz...  
32   Já tirou sua inscrição do MBL? \nAmo meu capitão  
..                                                ...  
39  Um ícone de simpatia  dotado de extrema simpli...  
38                  Abraços Presidente vc é um Fofo 😊  
37  DIA 26/05, TODO O POVO DE BEM NAS RUAS DO BRAS...  
35  Legal!\n O cara está mais a vontade com o carg...  
99                   Além de honesto é divertido\n😂😂😂  

[100 rows x 6 columns]

## Primeiras descobertas

- canal (poop) seguido por usuário - Meu sonho virou realidade ? Stand up do Bolsonaro? - Richteru - https://www.youtube.com/channel/UCYKolLcvVjavQN64-gRP9Sg

- prováveis bots - Joao Vitor http://www.youtube.com/channel/UCJoRyy9av5tnUZguUasNt3g

- comentários '2019-05-25' -próximos de manifestação que seria no (26) 

- Vídeo com mais comentário se refere ao dia 26 (um canal meio estranho)

 > 2019-05-24T01:54:47Z - Marcus Vinícius http://www.youtube.com/channel/UC01Y2iS0xR8vcTuNR1vb-Vw 272 -L/ 21 -R/ 
 Bolsonaro,  estarei na rua dia 26 pro seu Governo e pelo nosso amado Brasil. id = # UgyVEn8lUWNigBLcUGF4AaABAg

- Os comentários foram povoados por lado contrário "Ninguém Solta a Mão de Ninguém Ninguém Solta a Mão de Ninguém"- https://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg


## Respostas ao comentário 1

autor: Ninguém Solta a Mão de Ninguém (http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg ) 
 -@Anderson Silva  o Brasil anseia por isso...bozo se suicide ja.. ja era..ja foi...derreteu o apoio ao lixo...va estudar - 

autor: Ninguém Solta a Mão de Ninguém (http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg ) 
 -@Zazá Torres  o Brasil anseia por isso...bozo se suicide ja.. ja era..ja foi...derreteu o apoio ao lixo...va estudar - 

autor: Ninguém Solta a Mão de Ninguém (http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg ) 
 -@Oracy Ribeiro Mueller  o Brasil anseia por isso...bozo se suicide ja.. ja era..ja foi...derreteu o apoio ao lixo...va estudar - 

autor: Ninguém Solta a Mão de Ninguém (http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg ) 
 -@Double vala o Brasil snseia pelo suicídio do bozoluxo...o apoio a ele derreteu...fui - 

autor: Oracy Ribeiro Mueller (http://www.youtube.com/channel/UC7Rlw_REBuyanKXNHlvkGjQ ) 
 -@Ninguém Solta a Mão de Ninguém teu cú demônio, volta pro inferno desgraçado. - 

autor: Oracy Ribeiro Mueller (http://www.youtube.com/channel/UC7Rlw_REBuyanKXNHlvkGjQ ) 
 -@Ninguém Solta a Mão de Ninguém satanUsta??? Kkkkk vc é um jumento mesmo né queridinha. - 

autor: Zazá Torres (http://www.youtube.com/channel/UCwYkhzI-ObJGvLfZX5FBTxA ) 
 -@Ninguém Solta a Mão de Ninguém Vai pra casa do caralho, antes que eu me esqueça, seu verme - 

autor: Anderson Silva (http://www.youtube.com/channel/UC182I5PGSYZlwS7JFBlulRg ) 
 -@Ninguém Solta a Mão de Ninguém BABACA - 

autor: Ivanildo Queros Ferreira (http://www.youtube.com/channel/UCJ1ODMyHIW8NTctnYwKCjuw ) 
 -Tamos juntos! Presidente Bolsonaro, cada dia melhor!!! - 

autor: Vinicius Silva (http://www.youtube.com/channel/UCFYKCPJwtNMzMs92D6oOzAQ ) 
 -@Ninguém Solta a Mão de Ninguém deve ser um moleque catarrento ou um maconheiro da USP tnc morre que passa amigo, ou se mude para a Venezuela lá é melhor pra vc!!!!!!!!! - 

autor: Marcus Vinícius (http://www.youtube.com/channel/UC01Y2iS0xR8vcTuNR1vb-Vw ) 
 -@Ninguém Solta a Mão de Ninguém Como assim orar pela morte de alguém? Sinceramente nao consegui entender. - 

autor: aline almeida (http://www.youtube.com/channel/UCO5EdMxO0RdsF2H9_gMXODQ ) 
 -@Ninguém Solta a Mão de Ninguém Seu nome é Brasil? Vc foi nomeado porta voz da maioria dos brasileiros? Quanto ódio hem??? - 

autor: Kamila F (http://www.youtube.com/channel/UCY6BLe1uf7Fco5rK8IDv7ug ) 
 -@Ninguém Solta a Mão de Ninguém oro pelo Brasil, e tbm vou orar por vc. - 

autor: Gabriel Ferreira (http://www.youtube.com/channel/UCK_tIZVv5n50VA95btxXT8Q ) 
 -@Ninguém Solta a Mão de Ninguém tbm vou orar por vc - 

autor: Ninguém Solta a Mão de Ninguém (http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg ) 
 -@Marcus Vinícius caro....ore pelo brasil..ore pela morte do bozoluxo....o Brasil precisa dele no inferno de volta..fui - 

autor: Marcus Vinícius (http://www.youtube.com/channel/UC01Y2iS0xR8vcTuNR1vb-Vw ) 
 -@Ninguém Solta a Mão de Ninguém Vou orar pra você mais uma vez!!! - 

autor: Ninguém Solta a Mão de Ninguém (http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg ) 
 -@Marcus Vinícius bozo..ou renuncia.ou se mata...ou...cairs..demoniaco..satanusta..vcs foram enganados..trouxas... - 

autor: Marcus Vinícius (http://www.youtube.com/channel/UC01Y2iS0xR8vcTuNR1vb-Vw ) 
 -@Ninguém Solta a Mão de Ninguém SUICIDAR??? KKKKKK ....Vou orar pra você!!! - 

autor: Ninguém Solta a Mão de Ninguém (http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg ) 
 -suicide-se bozolixooooo...ou renuncie..o Brasil anseia por isso. - 

autor: Marcus Vinícius (http://www.youtube.com/channel/UC01Y2iS0xR8vcTuNR1vb-Vw ) 
 -@Ly Bertolli O Brasil precisa de nós. Tem que dar muita gente pra anular a força do Congresso, se as manifestações forem devagar , o Congresso ganhará força. - 

autor: Ly Bertolli (http://www.youtube.com/channel/UCBhRpRlpS-TNhC2pUiD5VaQ ) 
 -Melhor todo mundo ficar de olho aberto, congresso tá querendo calar a gente e diminuir as manifestações do dia 26/05! -

## Respostas ao comentário 2

autor: Oracy Ribeiro Mueller (http://www.youtube.com/channel/UC7Rlw_REBuyanKXNHlvkGjQ ) 
 -Gente denunciem o comentário desse tal de Apb agenda progressista. - 

autor: Ivanildo Queros Ferreira (http://www.youtube.com/channel/UCJ1ODMyHIW8NTctnYwKCjuw ) 
 -Presidente Bolsonaro, cada dia melhor!!! - 

autor: Zizao __ (http://www.youtube.com/channel/UCsoabgAi7qpLDTdzP5A9fgQ ) 
 -@Ninguém Solta a Mão de Ninguém kkkkkkkkkkkkk - 

autor: Marinês Rodrigues (http://www.youtube.com/channel/UCZZUz5wDiVwJqG5c__5IRMA ) 
 -@Ly Bertolli já preparei meus cartazes, lavei e passei minha bandeira e a camisa verde e amarela 🇧🇷😎👉 - 

autor: Maria Terezinha (http://www.youtube.com/channel/UC2V3bZZGiO2B2ff44bC3IWQ ) 
 -@Virginia Maria Correto....todos que nao estao satisfeitos podem ir para estes países que quem governa é o comunismo..sejam felizes!! - 

autor: Virginia Maria (http://www.youtube.com/channel/UCF3uJdcyT-aeymEXwubvZkQ ) 
 -@Ninguém Solta a Mão de Ninguém vai p Venezuela. Pq o Brasil agr com esse presidente vai prosperar. - 

autor: Maria Terezinha (http://www.youtube.com/channel/UC2V3bZZGiO2B2ff44bC3IWQ ) 
 -@Ninguém Solta a Mão de Ninguém NÃO..O BRASIL ANCEIA QUE SEU LULA LADRÃO NÃO SAI DA CADEIA.E QUE A PROMESSA QUE DEUS FEZ NA VIDA DESTE HOMEM..SE CUMPRE...EM NOME DE JESUS!!AVANTE BRASIL COM A VERDADE. - 

autor: Ninguém Solta a Mão de Ninguém (http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg ) 
 -suicide-se bozolixooooo...ou renuncie..o Brasil anseia por isso. - 

autor: Meu Amor (http://www.youtube.com/channel/UCobThgEPdt0f1b0n_S04UxA ) 
 -@Ly Bertolli se estão votando é porque estão com medo... agora é que tenho mais motivos pra ir😏 - 

autor: Ly Bertolli (http://www.youtube.com/channel/UCBhRpRlpS-TNhC2pUiD5VaQ ) 
 -Galera, fiquem ligados, o congresso tá colocando as pautas do governo em votação urgente pra diminuir as manifestações do dia 26/05, não caiam nessa, vamos às ruas cobrar esses que só querem parar o país! - 

autor: Cai Nessa (http://www.youtube.com/channel/UCnhZcQmEk8mZ-9Iu7EZwJGA ) 
 -@Mikuo Desafortunado oq q tera? - 

autor: Mikuo Desafortunado (http://www.youtube.com/channel/UCFZB8jwI5K0NUTy7lK7GuPQ ) 
 -galera liguem na record tv

In [185]:
comment_J003['items'][5]

{'kind': 'youtube#commentThread',
 'etag': 'X1PK6-cAwuYSJnDosQ3lJPZDJx0',
 'id': 'Ugyti96BYoVN2pWYcHd4AaABAg',
 'snippet': {'videoId': 'jg47rYougAs',
  'topLevelComment': {'kind': 'youtube#comment',
   'etag': 'R8b5woMZYnTNQTDjOLOFYP2iNCE',
   'id': 'Ugyti96BYoVN2pWYcHd4AaABAg',
   'snippet': {'videoId': 'jg47rYougAs',
    'textDisplay': 'Meu sonho virou realidade ? Stand up do Bolsonaro?',
    'textOriginal': 'Meu sonho virou realidade ? Stand up do Bolsonaro?',
    'authorDisplayName': 'Richteru',
    'authorProfileImageUrl': 'https://yt3.ggpht.com/a/AATXAJzL2sSUh0j3yLroPO32uhBWMuphxqDvm66SuPiZ=s48-c-k-c0xffffffff-no-rj-mo',
    'authorChannelUrl': 'http://www.youtube.com/channel/UCYKolLcvVjavQN64-gRP9Sg',
    'authorChannelId': {'value': 'UCYKolLcvVjavQN64-gRP9Sg'},
    'canRate': True,
    'viewerRating': 'none',
    'likeCount': 828,
    'publishedAt': '2019-05-24T01:53:34Z',
    'updatedAt': '2019-05-24T01:53:34Z'}},
  'canReply': True,
  'totalReplyCount': 12,
  'isPublic': True

In [152]:
q_commentR = youtube.comments().list(parentId ='UgzFm0bvyFpRmweeNyx4AaABAg', part = "snippet", maxResults = 100)

reply = q_commentR.execute()

In [160]:
reply['items'][0]

{'kind': 'youtube#comment',
 'etag': '22ytcfT8wXw4VYevcdIT5dsLafY',
 'id': 'UgzFm0bvyFpRmweeNyx4AaABAg.8vITd3O7n0W8vJRLWFh2zj',
 'snippet': {'textDisplay': '@Anderson Silva  o Brasil anseia por isso...bozo se suicide ja.. ja era..ja foi...derreteu o apoio ao lixo...va estudar',
  'textOriginal': '@Anderson Silva  o Brasil anseia por isso...bozo se suicide ja.. ja era..ja foi...derreteu o apoio ao lixo...va estudar',
  'parentId': 'UgzFm0bvyFpRmweeNyx4AaABAg',
  'authorDisplayName': 'Ninguém Solta a Mão de Ninguém',
  'authorProfileImageUrl': 'https://yt3.ggpht.com/a/AATXAJw_z6j9rReiVE9Vbk708CTYcLhEBFc9dGYBLHwgqw=s48-c-k-c0xffffffff-no-rj-mo',
  'authorChannelUrl': 'http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg',
  'authorChannelId': {'value': 'UCoLImFFtvwnNY3L24-dRqVg'},
  'canRate': True,
  'viewerRating': 'none',
  'likeCount': 0,
  'publishedAt': '2019-05-24T10:54:01Z',
  'updatedAt': '2019-05-24T10:54:01Z'}}

In [192]:
for r in reply['items']:
    print("autor: "+r['snippet']['authorDisplayName'],
          "(" +r['snippet']['authorChannelUrl'], ")", "\n",
          "-"+r['snippet']['textOriginal'],"-", "\n")

autor: Ninguém Solta a Mão de Ninguém (http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg ) 
 -@Anderson Silva  o Brasil anseia por isso...bozo se suicide ja.. ja era..ja foi...derreteu o apoio ao lixo...va estudar - 

autor: Ninguém Solta a Mão de Ninguém (http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg ) 
 -@Zazá Torres  o Brasil anseia por isso...bozo se suicide ja.. ja era..ja foi...derreteu o apoio ao lixo...va estudar - 

autor: Ninguém Solta a Mão de Ninguém (http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg ) 
 -@Oracy Ribeiro Mueller  o Brasil anseia por isso...bozo se suicide ja.. ja era..ja foi...derreteu o apoio ao lixo...va estudar - 

autor: Ninguém Solta a Mão de Ninguém (http://www.youtube.com/channel/UCoLImFFtvwnNY3L24-dRqVg ) 
 -@Double vala o Brasil snseia pelo suicídio do bozoluxo...o apoio a ele derreteu...fui - 

autor: Oracy Ribeiro Mueller (http://www.youtube.com/channel/UC7Rlw_REBuyanKXNHlvkGjQ ) 
 -@Ninguém Solta a Mão de Ninguém teu cú demô

In [190]:
q_commentR2 = youtube.comments().list(parentId ='Ugyti96BYoVN2pWYcHd4AaABAg', part = "snippet", maxResults = 100)

reply2 = q_commentR2.execute()

In [193]:
for r in reply2['items']:
    print("autor: "+r['snippet']['authorDisplayName'],
          "(" +r['snippet']['authorChannelUrl'], ")", "\n",
          "-"+r['snippet']['textOriginal'],"-", "\n")

autor: Oracy Ribeiro Mueller (http://www.youtube.com/channel/UC7Rlw_REBuyanKXNHlvkGjQ ) 
 -Gente denunciem o comentário desse tal de Apb agenda progressista. - 

autor: Ivanildo Queros Ferreira (http://www.youtube.com/channel/UCJ1ODMyHIW8NTctnYwKCjuw ) 
 -Presidente Bolsonaro, cada dia melhor!!! - 

autor: Zizao __ (http://www.youtube.com/channel/UCsoabgAi7qpLDTdzP5A9fgQ ) 
 -@Ninguém Solta a Mão de Ninguém kkkkkkkkkkkkk - 

autor: Marinês Rodrigues (http://www.youtube.com/channel/UCZZUz5wDiVwJqG5c__5IRMA ) 
 -@Ly Bertolli já preparei meus cartazes, lavei e passei minha bandeira e a camisa verde e amarela 🇧🇷😎👉 - 

autor: Maria Terezinha (http://www.youtube.com/channel/UC2V3bZZGiO2B2ff44bC3IWQ ) 
 -@Virginia Maria Correto....todos que nao estao satisfeitos podem ir para estes países que quem governa é o comunismo..sejam felizes!! - 

autor: Virginia Maria (http://www.youtube.com/channel/UCF3uJdcyT-aeymEXwubvZkQ ) 
 -@Ninguém Solta a Mão de Ninguém vai p Venezuela. Pq o Brasil agr com es